{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aba0374e-6279-4d01-9fc3-883963ef4203",
   "metadata": {},
   "source": [
    "# Instalasi dan Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7a27954f-27ef-42f3-a03f-3bbfba77ba9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: openai in /opt/anaconda3/lib/python3.11/site-packages (0.27.0)\n",
      "Requirement already satisfied: requests>=2.20 in /opt/anaconda3/lib/python3.11/site-packages (from openai) (2.32.3)\n",
      "Requirement already satisfied: tqdm in /opt/anaconda3/lib/python3.11/site-packages (from openai) (4.66.5)\n",
      "Requirement already satisfied: aiohttp in /opt/anaconda3/lib/python3.11/site-packages (from openai) (3.9.3)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /opt/anaconda3/lib/python3.11/site-packages (from requests>=2.20->openai) (2.0.4)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /opt/anaconda3/lib/python3.11/site-packages (from requests>=2.20->openai) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/anaconda3/lib/python3.11/site-packages (from requests>=2.20->openai) (2.0.7)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /opt/anaconda3/lib/python3.11/site-packages (from requests>=2.20->openai) (2024.2.2)\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in /opt/anaconda3/lib/python3.11/site-packages (from aiohttp->openai) (1.2.0)\n",
      "Requirement already satisfied: attrs>=17.3.0 in /opt/anaconda3/lib/python3.11/site-packages (from aiohttp->openai) (23.1.0)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in /opt/anaconda3/lib/python3.11/site-packages (from aiohttp->openai) (1.4.0)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in /opt/anaconda3/lib/python3.11/site-packages (from aiohttp->openai) (6.0.4)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in /opt/anaconda3/lib/python3.11/site-packages (from aiohttp->openai) (1.9.3)\n"
     ]
    }
   ],
   "source": [
    "!pip install openai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "38c1cdb9-3f9d-4f48-9aca-a011e02befbc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-10-28 01:32:08,903\tINFO worker.py:1816 -- Started a local Ray instance.\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "23a18851a133422faaaeff201b5f96df",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<div class=\"lm-Widget p-Widget lm-Panel p-Panel jp-Cell-outputWrapper\">\n",
       "    <div style=\"margin-left: 50px;display: flex;flex-direction: row;align-items: center\">\n",
       "        <div class=\"jp-RenderedHTMLCommon\" style=\"display: flex; flex-direction: row;\">\n",
       "  <svg viewBox=\"0 0 567 224\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" style=\"height: 3em;\">\n",
       "    <g clip-path=\"url(#clip0_4338_178347)\">\n",
       "        <path d=\"M341.29 165.561H355.29L330.13 129.051C345.63 123.991 354.21 112.051 354.21 94.2307C354.21 71.3707 338.72 58.1807 311.88 58.1807H271V165.561H283.27V131.661H311.8C314.25 131.661 316.71 131.501 319.01 131.351L341.25 165.561H341.29ZM283.29 119.851V70.0007H311.82C331.3 70.0007 342.34 78.2907 342.34 94.5507C342.34 111.271 331.34 119.861 311.82 119.861L283.29 119.851ZM451.4 138.411L463.4 165.561H476.74L428.74 58.1807H416L367.83 165.561H380.83L392.83 138.411H451.4ZM446.19 126.601H398L422 72.1407L446.24 126.601H446.19ZM526.11 128.741L566.91 58.1807H554.35L519.99 114.181L485.17 58.1807H472.44L514.01 129.181V165.541H526.13V128.741H526.11Z\" fill=\"var(--jp-ui-font-color0)\"/>\n",
       "        <path d=\"M82.35 104.44C84.0187 97.8827 87.8248 92.0678 93.1671 87.9146C98.5094 83.7614 105.083 81.5067 111.85 81.5067C118.617 81.5067 125.191 83.7614 130.533 87.9146C135.875 92.0678 139.681 97.8827 141.35 104.44H163.75C164.476 101.562 165.622 98.8057 167.15 96.2605L127.45 56.5605C121.071 60.3522 113.526 61.6823 106.235 60.3005C98.9443 58.9187 92.4094 54.9203 87.8602 49.0574C83.3109 43.1946 81.0609 35.8714 81.5332 28.4656C82.0056 21.0599 85.1679 14.0819 90.4252 8.8446C95.6824 3.60726 102.672 0.471508 110.08 0.0272655C117.487 -0.416977 124.802 1.86091 130.647 6.4324C136.493 11.0039 140.467 17.5539 141.821 24.8501C143.175 32.1463 141.816 39.6859 138 46.0505L177.69 85.7505C182.31 82.9877 187.58 81.4995 192.962 81.4375C198.345 81.3755 203.648 82.742 208.33 85.3976C213.012 88.0532 216.907 91.9029 219.616 96.5544C222.326 101.206 223.753 106.492 223.753 111.875C223.753 117.258 222.326 122.545 219.616 127.197C216.907 131.848 213.012 135.698 208.33 138.353C203.648 141.009 198.345 142.375 192.962 142.313C187.58 142.251 182.31 140.763 177.69 138L138 177.7C141.808 184.071 143.155 191.614 141.79 198.91C140.424 206.205 136.44 212.75 130.585 217.313C124.731 221.875 117.412 224.141 110.004 223.683C102.596 223.226 95.6103 220.077 90.3621 214.828C85.1139 209.58 81.9647 202.595 81.5072 195.187C81.0497 187.779 83.3154 180.459 87.878 174.605C92.4405 168.751 98.9853 164.766 106.281 163.401C113.576 162.035 121.119 163.383 127.49 167.19L167.19 127.49C165.664 124.941 164.518 122.182 163.79 119.3H141.39C139.721 125.858 135.915 131.673 130.573 135.826C125.231 139.98 118.657 142.234 111.89 142.234C105.123 142.234 98.5494 139.98 93.2071 135.826C87.8648 131.673 84.0587 125.858 82.39 119.3H60C58.1878 126.495 53.8086 132.78 47.6863 136.971C41.5641 141.163 34.1211 142.972 26.7579 142.059C19.3947 141.146 12.6191 137.574 7.70605 132.014C2.79302 126.454 0.0813599 119.29 0.0813599 111.87C0.0813599 104.451 2.79302 97.2871 7.70605 91.7272C12.6191 86.1673 19.3947 82.5947 26.7579 81.6817C34.1211 80.7686 41.5641 82.5781 47.6863 86.7696C53.8086 90.9611 58.1878 97.2456 60 104.44H82.35ZM100.86 204.32C103.407 206.868 106.759 208.453 110.345 208.806C113.93 209.159 117.527 208.258 120.522 206.256C123.517 204.254 125.725 201.276 126.771 197.828C127.816 194.38 127.633 190.677 126.253 187.349C124.874 184.021 122.383 181.274 119.205 179.577C116.027 177.88 112.359 177.337 108.826 178.042C105.293 178.746 102.113 180.654 99.8291 183.44C97.5451 186.226 96.2979 189.718 96.3 193.32C96.2985 195.364 96.7006 197.388 97.4831 199.275C98.2656 201.163 99.4132 202.877 100.86 204.32ZM204.32 122.88C206.868 120.333 208.453 116.981 208.806 113.396C209.159 109.811 208.258 106.214 206.256 103.219C204.254 100.223 201.275 98.0151 197.827 96.97C194.38 95.9249 190.676 96.1077 187.348 97.4873C184.02 98.8669 181.274 101.358 179.577 104.536C177.879 107.714 177.337 111.382 178.041 114.915C178.746 118.448 180.653 121.627 183.439 123.911C186.226 126.195 189.717 127.443 193.32 127.44C195.364 127.443 197.388 127.042 199.275 126.259C201.163 125.476 202.878 124.328 204.32 122.88ZM122.88 19.4205C120.333 16.8729 116.981 15.2876 113.395 14.9347C109.81 14.5817 106.213 15.483 103.218 17.4849C100.223 19.4868 98.0146 22.4654 96.9696 25.9131C95.9245 29.3608 96.1073 33.0642 97.4869 36.3922C98.8665 39.7202 101.358 42.4668 104.535 44.1639C107.713 45.861 111.381 46.4036 114.914 45.6992C118.447 44.9949 121.627 43.0871 123.911 40.301C126.195 37.515 127.442 34.0231 127.44 30.4205C127.44 28.3772 127.038 26.3539 126.255 24.4664C125.473 22.5788 124.326 20.8642 122.88 19.4205ZM19.42 100.86C16.8725 103.408 15.2872 106.76 14.9342 110.345C14.5813 113.93 15.4826 117.527 17.4844 120.522C19.4863 123.518 22.4649 125.726 25.9127 126.771C29.3604 127.816 33.0638 127.633 36.3918 126.254C39.7198 124.874 42.4664 122.383 44.1635 119.205C45.8606 116.027 46.4032 112.359 45.6988 108.826C44.9944 105.293 43.0866 102.114 40.3006 99.8296C37.5145 97.5455 34.0227 96.2983 30.42 96.3005C26.2938 96.3018 22.337 97.9421 19.42 100.86ZM100.86 100.86C98.3125 103.408 96.7272 106.76 96.3742 110.345C96.0213 113.93 96.9226 117.527 98.9244 120.522C100.926 123.518 103.905 125.726 107.353 126.771C110.8 127.816 114.504 127.633 117.832 126.254C121.16 124.874 123.906 122.383 125.604 119.205C127.301 116.027 127.843 112.359 127.139 108.826C126.434 105.293 124.527 102.114 121.741 99.8296C118.955 97.5455 115.463 96.2983 111.86 96.3005C109.817 96.299 107.793 96.701 105.905 97.4835C104.018 98.2661 102.303 99.4136 100.86 100.86Z\" fill=\"#00AEEF\"/>\n",
       "    </g>\n",
       "    <defs>\n",
       "        <clipPath id=\"clip0_4338_178347\">\n",
       "            <rect width=\"566.93\" height=\"223.75\" fill=\"white\"/>\n",
       "        </clipPath>\n",
       "    </defs>\n",
       "  </svg>\n",
       "</div>\n",
       "\n",
       "        <table class=\"jp-RenderedHTMLCommon\" style=\"border-collapse: collapse;color: var(--jp-ui-font-color1);font-size: var(--jp-ui-font-size1);\">\n",
       "    <tr>\n",
       "        <td style=\"text-align: left\"><b>Python version:</b></td>\n",
       "        <td style=\"text-align: left\"><b>3.11.5</b></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td style=\"text-align: left\"><b>Ray version:</b></td>\n",
       "        <td style=\"text-align: left\"><b>2.38.0</b></td>\n",
       "    </tr>\n",
       "    \n",
       "</table>\n",
       "\n",
       "    </div>\n",
       "</div>\n"
      ],
      "text/plain": [
       "RayContext(dashboard_url='', python_version='3.11.5', ray_version='2.38.0', ray_commit='385ee466260ef3cd218d5e372aef5d39338b7b94')"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !pip install ray transformers sentence-transformers torch pytesseract Pillow\n",
    "#!pip install --upgrade \"jax[cpu]\"\n",
    "\n",
    "import ray\n",
    "import time\n",
    "from transformers import T5ForConditionalGeneration, T5Tokenizer\n",
    "from sentence_transformers import SentenceTransformer, util\n",
    "from Bio import Entrez\n",
    "import pytesseract\n",
    "from PIL import Image\n",
    "import os\n",
    "import torch\n",
    "import warnings\n",
    "import re\n",
    "import pandas as pd\n",
    "from typing import List\n",
    "\n",
    "# Set environment variables\n",
    "email_address = \"dwiyulianto31072k4@gmail.com\"\n",
    "os.environ[\"TOKENIZERS_PARALLELISM\"] = \"false\"\n",
    "\n",
    "# Menonaktifkan FutureWarning terkait resume_download\n",
    "warnings.simplefilter(action='ignore', category=FutureWarning)\n",
    "\n",
    "# Inisialisasi Ray\n",
    "ray.shutdown()\n",
    "ray.init()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfbeabc2-220e-4635-b9c0-22e6d6f3fc7c",
   "metadata": {},
   "source": [
    "# Agent Pencarian Data PubMed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0a1f3cdf-77db-4a2e-85bf-e6cc9be6b646",
   "metadata": {},
   "outputs": [],
   "source": [
    "@ray.remote\n",
    "class DataRetrievalAgent:\n",
    "    def __init__(self):\n",
    "        Entrez.email = email_address\n",
    "\n",
    "    def search_pubmed(self, keywords, max_results=5):\n",
    "        search_results = []\n",
    "        for term in keywords:\n",
    "            handle = Entrez.esearch(db=\"pubmed\", term=term, retmax=max_results)\n",
    "            record = Entrez.read(handle)\n",
    "            pmids = record[\"IdList\"]\n",
    "            search_results.extend(pmids)\n",
    "            handle.close()\n",
    "        return search_results\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4585de42-393a-471a-9914-ed5019be84ac",
   "metadata": {},
   "source": [
    "# Agent Ekstraksi Abstrak dari PubMed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ac8ac121-c310-407d-a50a-a1992de70305",
   "metadata": {},
   "outputs": [],
   "source": [
    "@ray.remote\n",
    "class ExtractionAgent:\n",
    "    def __init__(self):\n",
    "        Entrez.email = email_address\n",
    "\n",
    "    def fetch_abstract(self, pmid):\n",
    "        handle = Entrez.efetch(db=\"pubmed\", id=pmid, rettype=\"abstract\", retmode=\"text\")\n",
    "        abstract = handle.read()\n",
    "        handle.close()\n",
    "        return abstract"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c63eb98-5f71-4b64-88e1-17442c77d646",
   "metadata": {},
   "source": [
    "# Agent Peringkasan Teks Abstrak Menggunakan Model T5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "eff3f800-f846-445e-92f5-2be81812518b",
   "metadata": {},
   "outputs": [],
   "source": [
    "@ray.remote\n",
    "class SummarizationAgent:\n",
    "    def __init__(self):\n",
    "        self.model = T5ForConditionalGeneration.from_pretrained(\"t5-base\")\n",
    "        self.tokenizer = T5Tokenizer.from_pretrained(\"t5-base\", model_max_length=512)\n",
    "    \n",
    "    def summarize_text(self, text):\n",
    "        input_text = \"summarize: \" + text\n",
    "        inputs = self.tokenizer(input_text, return_tensors=\"pt\", max_length=512, truncation=True)\n",
    "        summary_ids = self.model.generate(inputs.input_ids, max_length=150, min_length=30, length_penalty=2.0, num_beams=4, early_stopping=True)\n",
    "        return self.tokenizer.decode(summary_ids[0], skip_special_tokens=True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6406cc4-5050-45dd-8128-6df84de76ba8",
   "metadata": {},
   "source": [
    "# Agent OCR untuk Ekstraksi Teks dari Gambar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d0f0d46c-6bba-4032-9b70-8fc9ed70b847",
   "metadata": {},
   "outputs": [],
   "source": [
    "@ray.remote\n",
    "class OCRAgent:\n",
    "    def extract_text_from_images(self, folder_path):\n",
    "        ocr_texts = []\n",
    "        for filename in os.listdir(folder_path):\n",
    "            if filename.endswith(('.png', '.jpg', '.jpeg', '.tiff', '.bmp', '.gif')):\n",
    "                image_path = os.path.join(folder_path, filename)\n",
    "                text = pytesseract.image_to_string(Image.open(image_path))\n",
    "                ocr_texts.append(text)\n",
    "        return \" \".join(ocr_texts)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "903d4c77-640c-4a60-bdb2-3e399970cfd8",
   "metadata": {},
   "source": [
    "# Agent Pencocokan Relevansi Abstrak"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "42e369df-3212-4c5b-802e-3c742ff5032d",
   "metadata": {},
   "outputs": [],
   "source": [
    "@ray.remote\n",
    "class RelevanceAgent:\n",
    "    def __init__(self):\n",
    "        self.embedder = SentenceTransformer('all-MiniLM-L6-v2')\n",
    "    \n",
    "    def find_relevant_abstracts(self, abstracts, query, top_k=5):\n",
    "        abstract_embeddings = [self.embedder.encode(abstract, convert_to_tensor=True) for abstract in abstracts]\n",
    "        abstract_embeddings = torch.stack(abstract_embeddings)\n",
    "        query_embedding = self.embedder.encode(query, convert_to_tensor=True)\n",
    "        cosine_scores = util.pytorch_cos_sim(query_embedding, abstract_embeddings)\n",
    "        top_results = torch.topk(cosine_scores, k=top_k)\n",
    "        return [abstracts[i] for i in top_results[1][0].tolist()]\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e64eda6d-44b5-4330-aa14-b2606b03f606",
   "metadata": {},
   "source": [
    "# Fungsi Normalisasi Teks OCR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0a12766e-9f59-4f56-99f4-b61e51233ba5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def normalisasi_abstrak(teks: str) -> str:\n",
    "    teks = re.sub(r'\\\\bduosnueyy\\\\b', '', teks)  # contoh hapus kata 'duosnueyy' jika itu adalah noise OCR\n",
    "    teks = re.sub(r'\\\\s{2,}', ' ', teks)  # menghilangkan spasi berlebih\n",
    "    teks = re.sub(r'\\\\n+', ' ', teks)  # mengubah newline menjadi spasi\n",
    "    teks = re.sub(r'\\\\[.*?\\\\]', '', teks)  # hapus teks dalam tanda kurung kotak\n",
    "    return teks.strip()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85a2fcf3-73e8-4a22-a8b9-ecc7cb0c125c",
   "metadata": {},
   "source": [
    "# Fungsi Menyimpan Abstrak ke CSV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "57a1c454-1511-4227-932a-a46b781519c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def simpan_abstrak_ke_csv(abstrak_list: List[str], filename: str):\n",
    "    data = {'Abstrak': abstrak_list}\n",
    "    df = pd.DataFrame(data)\n",
    "    df.to_csv(filename, index=False)\n",
    "    print(f\"Abstrak yang relevan telah disimpan ke {filename}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7de6dcd-c42a-4a18-87ee-562d0cb1fef8",
   "metadata": {},
   "source": [
    "# Pipeline untuk Menjalankan Semua Agen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "77fb7c9c-ce51-4e73-9c70-e2607f7f5571",
   "metadata": {},
   "outputs": [],
   "source": [
    "def multi_agent_pipeline(search_terms, user_query, folder_path=None):\n",
    "    # Inisialisasi agen\n",
    "    retrieval_agent = DataRetrievalAgent.remote()\n",
    "    extraction_agent = ExtractionAgent.remote()\n",
    "    summarization_agent = SummarizationAgent.remote()\n",
    "    relevance_agent = RelevanceAgent.remote()\n",
    "    \n",
    "    # Langkah 1: Cari PubMed berdasarkan kata kunci\n",
    "    pmids = ray.get(retrieval_agent.search_pubmed.remote(search_terms))\n",
    "    print(\"Daftar PMIDs yang ditemukan:\", pmids)\n",
    "    \n",
    "    # Langkah 2: Ekstraksi abstrak dari PubMed\n",
    "    abstracts = ray.get([extraction_agent.fetch_abstract.remote(pmid) for pmid in pmids])\n",
    "    print(\"Contoh abstrak:\", abstracts[0] if abstracts else \"Tidak ada abstrak ditemukan.\")\n",
    "    \n",
    "    # Langkah 3: Jalankan OCR untuk batch gambar jika ada folder gambar yang disediakan\n",
    "    if folder_path:\n",
    "        ocr_agent = OCRAgent.remote()\n",
    "        ocr_text = ray.get(ocr_agent.extract_text_from_images.remote(folder_path))\n",
    "        ocr_text_normalized = normalisasi_abstrak(ocr_text)\n",
    "        print(\"Teks OCR yang telah dinormalisasi:\", ocr_text_normalized)\n",
    "    \n",
    "        # Filter dan simpan hasil abstrak relevan\n",
    "        kata_kunci = ['diabetes', 'peripheral neuropathy', 'neuropathy', 'diabetic']\n",
    "        abstrak_relevan = [ocr_text_normalized] if any(kata in ocr_text_normalized.lower() for kata in kata_kunci) else []\n",
    "        \n",
    "        if abstrak_relevan:\n",
    "            simpan_abstrak_ke_csv(abstrak_relevan, 'abstrak_relevan.csv')\n",
    "        \n",
    "        print(\"Abstrak yang relevan dan terfilter:\")\n",
    "        for idx, abstrak in enumerate(abstrak_relevan, 1):\n",
    "            print(f\"{idx}. {abstrak}\\\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "986f9726-f817-4627-a4dd-b8e8491fa5c3",
   "metadata": {},
   "source": [
    "# Menjalankan Pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b8683c76-d4ba-42b4-b76d-cdd639cec850",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[36m(SummarizationAgent pid=24902)\u001b[0m /opt/anaconda3/lib/python3.11/site-packages/huggingface_hub/file_download.py:1142: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.\n",
      "\u001b[36m(SummarizationAgent pid=24902)\u001b[0m   warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Daftar PMIDs yang ditemukan: ['39460909', '39460375', '39459574', '39459480', '39459455', '39460977', '39460909', '39460896', '39460887', '39460283', '39458946', '39458672', '39458583', '39454900', '39452591']\n",
      "Contoh abstrak: 1. Diabetes Ther. 2024 Oct 26. doi: 10.1007/s13300-024-01658-8. Online ahead of \n",
      "print.\n",
      "\n",
      "Current Perspectives in Pre- and Diabetic Peripheral Neuropathy Diagnosis and \n",
      "Management: An Expert Statement for the Gulf Region.\n",
      "\n",
      "Beshyah SA(1)(2)(3), Jayyousi A(4), Al-Mamari AS(5), Shaaban A(6), Ozairi EA(7), \n",
      "Nafach J(8), Jallo MKI(9), Khader S(10), Evans M(11).\n",
      "\n",
      "Author information:\n",
      "(1)Department of Medicine, NMC Royal, MBZ, Abu Dhabi, United Arab Emirates. \n",
      "beshyah@yahoo.com.\n",
      "(2)Department of Medicine, College of Medicine and Health Sciences, Khalifa \n",
      "University, Abu Dhabi, UAE. beshyah@yahoo.com.\n",
      "(3)Department of Medicine, Dubai Medical College, Dubai, United Arab Emirates. \n",
      "beshyah@yahoo.com.\n",
      "(4)Hamad Medical Corporation, Doha, Qatar.\n",
      "(5)Department of Medicine, Sultan Qaboos University Hospital, Muscat, Oman.\n",
      "(6)Diabetes Control Centre, Ghassan Najib Pharaon Hospital, Jeddah, Kingdom of \n",
      "Saudi Arabia.\n",
      "(7)Clinical Care Research and Clinical Trials Unit, Dasman Diabetes Institute, \n",
      "Dasman, Kuwait.\n",
      "(8)Department of Diabetes, Dubai Diabetes Center, Dubai, United Arab Emirates.\n",
      "(9)Internal Medicine and Endocrinology, Centre of Endocrinology, Gulf Medical \n",
      "University and Thumbay University Hospital, Ajman, United Arab Emirates.\n",
      "(10)Department of Diabetes and Endocrinology, Dr. Sulaiman Alhabib Medical \n",
      "Group, Riyadh, Saudi Arabia.\n",
      "(11)Diabetes Resource Center, University Hospital Llandough, Penarth, UK.\n",
      "\n",
      "Peripheral neuropathy (PN) significantly impacts the quality of life, causing \n",
      "substantial morbidity and increased mortality, as well as escalating healthcare \n",
      "costs. While PN can have various causes, the most common form, diabetic \n",
      "peripheral neuropathy, poses considerable risks for potential complications. \n",
      "Diabetic peripheral neuropathy (DPN) affects over 50% of people with prediabetes \n",
      "and diabetes. Despite its prevalence, a global gap in diagnosis and management \n",
      "exists, exacerbated by the COVID-19 pandemic. This expert consensus was \n",
      "formulated through a comprehensive evaluation by a panel of experts, informed by \n",
      "a focused literature review, aiming to establish a clinically robust approach to \n",
      "diagnosing and managing pre- and diabetic PN with the early utilization of \n",
      "neurotropic B vitamins. This document offers a consensus perspective on the \n",
      "existing challenges in diagnosing and managing PN, focusing on DPN. The expert \n",
      "panel proposes measures to address this underdiagnosed burden, highlighting the \n",
      "importance of early intervention through innovative screening methods, \n",
      "integrated care approaches, and therapeutic strategies. The document advocates \n",
      "for increased awareness, targeted campaigns, and proactive care strategies to \n",
      "bridge gaps in the patient care of individuals with diabetes, emphasizing the \n",
      "importance of early detection and timely management to improve overall health \n",
      "outcomes. Specific recommendations include incorporating simplified \n",
      "questionnaires and innovative screening methods into routine care, prioritizing \n",
      "neurotropic B vitamin supplementation, optimizing glucagon-like peptide 1 \n",
      "(GLP-1) receptor agonist treatments, and adopting a holistic approach to \n",
      "neuropathy management. The consensus underscores the urgent need to address the \n",
      "underdiagnosis and undertreatment of PN, offering practical measures to enhance \n",
      "early detection and improve health outcomes for individuals with DPN.\n",
      "\n",
      "© 2024. The Author(s).\n",
      "\n",
      "DOI: 10.1007/s13300-024-01658-8\n",
      "PMID: 39460909\n",
      "Teks OCR yang telah dinormalisasi: duosnueyy Jouiny yduosnueyy Jouny yduosnueyy Joujny\n",
      "\n",
      "duosnuey soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 14\n",
      "\n",
      "Supplementary Material\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "Acknowledgements\n",
      "\n",
      "This work was supported in part by a grant from the U.S. National Science Foundation (NSF CBET-096782). Susan\n",
      "Sharfstein was the recipient of a Fulbright Global Scholar Award. Prashant Kaushik is funded from the European\n",
      "Union’s Horizon 2020 research and innovation program under the Marie Sklodowska-Curie grant agreement No\n",
      "642663. The Orbitrap Fusion Tribrid mass spectrometer was funded under a Science Foundation Ireland\n",
      "Infrastructure Award, grant number 16/RI/3701. The RNA-Seq was funded with support from the Novo Nordisk\n",
      "Foundation provided to the Center for Biosustainability at the Technical University of Denmark\n",
      "(NNF10CC1016517), and analysis was funded by the U.S. National Institutes of Health NIGMS (R35 GM119850).\n",
      "The authors would like to thank Dr. Vandhana Muralidharan-Chari for performing the ELISA measurements.\n",
      "\n",
      "References\n",
      "\n",
      "Akopov SB, Ruda VM, Batrak VV, Vetchinova AS, Chernov IP, Nikolaev LG, Bode J, Sverdlov ED,\n",
      "2006 Identification, genome mapping, and CTCF binding of potential insulators within the FX YDS5-\n",
      "COX7A1 locus of human chromosome 19q13.12. Mamm. Genome 17, 1042-1049. 10.1007/\n",
      "s00335-006-0037-3. [PubMed: 17019650]\n",
      "\n",
      "Anders S, Pyl PT, Huber W, 2015 HTSeq-a Python framework to work with high-throughput\n",
      "sequencing data. Bioinformatics. 31, 166-169. 10.1093/bioinformatics/btu638. [PubMed:\n",
      "25260700]\n",
      "\n",
      "Beausoleil SA, Jedrychowski M, Schwartz D, Elias JE, Villén J, Li J, Cohn MA, Cantley LC, Gygi SP,\n",
      "2004 Large-scale characterization of HeLa cell nuclear phospho-proteins. Proc. Natl. Acad. Sci. U.\n",
      "S. A 101, 12130-12135. 10.1073/pnas.0404720101. [PubMed: 15302935]\n",
      "\n",
      "Brown AJ, Sweeney B, Mainwaring DO, James DC, 2014 Synthetic promoters for CHO cell\n",
      "engineering. Biotechnol. Bioeng 111, 1638-1647. 10.1002/bit.25227. [PubMed: 24615264]\n",
      "\n",
      "Brown AJ, Sweeney B, Mainwaring DO, James DC, 2015 NF-«B, CRE and YY1 elements are key\n",
      "functional regulators of CMV promoter-driven transient gene expression in CHO cells. Biotechnol.\n",
      "J 10, 1019-1028. 10.1002/biot.201400744. [PubMed: 25612069]\n",
      "\n",
      "Chiang AWT, Li S, Kellman BP, Chattopadhyay G, Zhang Y, Kuo C-C, Gutierrez JM, Ghazi F,\n",
      "Schmeisser H, Ménard P, Bjgrn SP, Voldborg BG, Rosenberg AS, Puig M, Lewis NE, 2019\n",
      "Combating viral contaminants in CHO cells by engineering innate immunity. Sci. Rep 9, 8827\n",
      "10.1038/s41598-019-45126-x. [PubMed: 31222165]\n",
      "\n",
      "Dahodwala H, Sharfstein ST, 2014 Role of epigenetics in expression of recombinant proteins from\n",
      "mammalian cells. Pharm. Bioprocess 2, 403-419. 10.4155/pbp.14.47.\n",
      "\n",
      "Dahodwala H, Sharfstein ST, 2017 In: Meleady P (Ed.), The ‘Omics Revolution in CHO Biology:\n",
      "Roadmap to Improved CHO Productivity BT - Heterologous Protein Production in CHO Cells:\n",
      "Methods and Protocols. Springer, New York, New York, NY, pp. 153-168.\n",
      "10.1007/978-1-4939-6972-2_10.\n",
      "\n",
      "Dahodwala H, Nowey M, Mitina T, Sharfstein ST, 2012 Effects of clonal variation on growth,\n",
      "metabolism, and productivity in response to trophic factor stimulation: a study of Chinese hamster\n",
      "ovary cells producing a recombinant monoclonal antibody. Cytotechnology. 64, 27-41. 10.1007/\n",
      "s10616-011-9388-z. [PubMed: 21822681]\n",
      "\n",
      "Dobin A, Davis CA, Schlesinger F, Drenkow J, Zaleski C, Jha S, Batut P, Chaisson M, Gingeras TR,\n",
      "2013 STAR: ultrafast universal RNA-seq aligner. Bioinformatics. 29, 15-21. 10.1093/\n",
      "bioinformatics/bts635. [PubMed: 23104886]\n",
      "\n",
      "Eng JK, McCormack AL, Yates JR, 1994 An approach to correlate tandem mass spectral data of\n",
      "peptides with amino acid sequences in a protein database. J. Am. Soc. Mass Spectrom 5, 976-989.\n",
      "10.1016/1044-0305(94)80016-2. [PubMed: 24226387]\n",
      "\n",
      "Feichtinger J, Hernandez I, Fischer C, Hanscho M, Auer N, Hackl M, Jadhav V, Baumann M, Krempl\n",
      "PM, Schmidl C, Farlik M, Schuster M, Merkel A, Sommer A, Heath S, Rico D, Bock C,\n",
      "Thallinger GG, Borth N, 2016 Comprehensive genome and epigenome characterization of CHO\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Joulny ydiuosnueyy Jouny duosnueyy Jouiny\n",
      "\n",
      "duosnuey soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 15\n",
      "\n",
      "cells in response to evolutionary pressures and over time. Biotechnol. Bioeng 113, 2241-2253.\n",
      "10.1002/bit.25990. [PubMed: 27072894]\n",
      "\n",
      "Harton JA, Zika E, Ting JP, 2001 The histone acetyltransferase domains of CREB-binding protein\n",
      "(CBP) and p300/CBP-associated factor are not necessary for cooperativity with the class II\n",
      "transactivator. J. Biol. Chem 276, 38715-38720. 10.1074/jbc.M 106652200. [PubMed: 11514574]\n",
      "\n",
      "Hausmann R, Chudobova I, Spiegel H, Schillberg S, 2018 Proteomic analysis of CHO cell lines\n",
      "producing high and low quantities of a recombinant antibody before and after selection with\n",
      "methotrexate. J. Biotechnol 10.1016/j.jbiotec.2017.11.008.\n",
      "\n",
      "Henry M, Power M, Kaushik P, Coleman O, Clynes M, Meleady P, 2017 Differential\n",
      "phosphoproteomic analysis of recombinant Chinese hamster ovary cells following temperature\n",
      "shift. J. Proteome Res 10.1021/acs.jproteome.6b00868acs.jproteome.6b00868.\n",
      "\n",
      "Hervouet E, Vallette FM, Cartron P-F, 2009 Dnmt3/transcription factor interactions as crucial players\n",
      "in targeted DNA methylation. Epigenetics. 4, 487-499. 10.4161/epi.4.7.9883. [PubMed:\n",
      "19786833]\n",
      "\n",
      "Ho SCL, Koh EYC, Soo BPC, Mariati S-H, Chao Y. Yang, 2016 Evaluating the use of a CpG free\n",
      "promoter for long-term recombinant protein expression stability in Chinese hamster ovary cells.\n",
      "BMC Biotechnol. 16, 71 10.1186/s12896-016-0300-y. [PubMed: 27756290]\n",
      "\n",
      "Jiang Z, Sharfstein ST, 2008 Sodium butyrate stimulates monoclonal antibody over-expression in CHO\n",
      "cells by improving gene accessibility. Biotechnol. Bioeng 100, 189-194. 10.1002/bit.21726.\n",
      "[PubMed: 18023047]\n",
      "\n",
      "Jiang Z, Sharfstein ST, 2009 Characterization of gene localization and accessibility in DHFR-\n",
      "amplified CHO cells. Biotechnol. Prog 25, 296-300. 10.1002/btpr.82. [PubMed: 19224609]\n",
      "\n",
      "Jiang Z, Huang Y, Sharfstein ST, 2006 Regulation of recombinant monoclonal antibody production in\n",
      "Chinese hamster ovary cells: a comparative study of gene copy number, mRNA level, and protein\n",
      "expression. Biotechnol. Prog 22, 313-318. 10.1021/bp0501524. [PubMed: 16454525]\n",
      "\n",
      "Kaushik P, Henry M, Clynes M, Meleady P, 2018 The expression pattern of the phosphoproteome is\n",
      "significantly changed during the growth phases of recombinant CHO cell culture. Biotechnol. J 0,\n",
      "1700221 10.1002/biot.201700221.\n",
      "\n",
      "Khan AS, Murray MJ, Ho CMK, Zuercher WJ, Reeves MB, Strang BL, 2017 High-throughput\n",
      "screening of a GlaxoSmithKline protein kinase inhibitor set identifies an inhibitor of human\n",
      "cytomegalovirus replication that prevents CREB and histone H3 post-translational modification. J.\n",
      "Gen. Virol 98, 754-768. 10.1099/jgv.0.000713. [PubMed: 28100301]\n",
      "\n",
      "Kim SH, Trinh AT, Larsen MC, Mastrocola AS, Jefcoate CR, Bushel PR, Tibbetts RS, 2016 Tunable\n",
      "regulation of CREB DNA binding activity couples genotoxic stress response and metabolism.\n",
      "Nucleic Acids Res. 10.1093/nar/gkw643.\n",
      "\n",
      "Kohler G, Milstein C, 1975 Continuous cultures of fused cells secreting antibody of predefined\n",
      "specificity. Nature. 256, 495-497. 10.1038/256495a0. [PubMed: 1172191]\n",
      "\n",
      "Kovacs KA, Steinmann M, Halfon O, Magistretti PJ, Cardinaux JR, 2015 Complex regulation of\n",
      "CREB-binding protein by homeodomain-interacting protein kinase 2. Cell. Signal 10.1016/\n",
      "j.cellsig.2015.08.001.\n",
      "\n",
      "Kramer A, Green J, Pollard J Jr., Tugendreich S, 2013 Causal analysis approaches in Ingenuity\n",
      "Pathway Analysis. Bioinformatics. 30, 523-530. 10.1093/bioinformatics/btt703. [PubMed:\n",
      "24336805]\n",
      "\n",
      "Levine AA, Guan Z, Barco A, Xu S, Kandel ER, Schwartz JH, 2005 CREB-binding protein controls\n",
      "response to cocaine by acetylating histones at the fosB promoter in the mouse striatum. Proc. Natl.\n",
      "Acad. Sci. U. S. A 102 10.1073/pnas.0509735102 19186 LP- 19191. [PubMed: 16380431]\n",
      "\n",
      "Lewis NE, Liu X, Li Y, Nagarajan H, Yerganian G, O’Brien E, Bordbar A, Roth AM, Rosenbloom J,\n",
      "Bian C, Xie M, Chen W, Li N, Baycin-Hizal D, Latif H, Forster J, Betenbaugh MJ, Famili I, Xu X,\n",
      "Wang J, Palsson BO, 2013 Genomic landscapes of Chinese hamster ovary cell lines as revealed by\n",
      "the Cricetulus griseus draft genome. Nat. Biotechnol 31, 759-767. 10.1038/nbt.2624. [PubMed:\n",
      "23873082]\n",
      "\n",
      "Li S, Cha SW, Heffner K, Hizal DB, Bowen MA, Chaerkady R, Cole RN, Tejwani V, Kaushik P,\n",
      "Henry M, Meleady P, Sharfstein ST, Betenbaugh MJ, Bafna V, Lewis NE, 2019 Proteogenomic\n",
      "annotation of Chinese hamsters reveals extensive novel translation events and endogenous\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " diosnueyy soulny ydiosnueyy Joylny yduosnueyy soulny\n",
      "\n",
      "yduosnueyy soulny\n",
      "\n",
      "\"heavaaa\n",
      "\n",
      "Author manuscript\n",
      "Cur Res Biotechnol, Author manuscript; available in PMC 2020 June 23.\n",
      "\n",
      "~“y HHS Public Access\n",
      "Ka\n",
      "\n",
      "Published in final edited form as:\n",
      "Curr Res Biotechnol. 2019 November ; 1: 49-57. doi:10.1016/j.crbiot.2019.09.001.\n",
      "\n",
      "Increased mAb production in amplified CHO cell lines is\n",
      "associated with increased interaction of CREB1 with transgene\n",
      "promoter\n",
      "\n",
      "Hussain Dahodwala®!:2, Prashant Kaushik®', Vijay Tejwani?, Chih-Chung Kuo‘, Patrice\n",
      "Menard‘, Michael Henry®, Bjorn G. Voldborg®, Nathan E. Lewis®:°, Paula Meleady®, Susan\n",
      "T. Sharfstein@”\n",
      "\n",
      "4College of Nanoscale Science and Engineering, SUNY Polytechnic Institute, Albany, NY, USA\n",
      "National Institute for Cellular Biotechnology, Dublin City University, Dublin 9, Ireland (Department\n",
      "of Bioengineering, University of California, San Diego, La Jolla, CA, USA “Novo Nordisk\n",
      "Foundation Center for Biosustainability, Technical University of Denmark, 2800 Kgs. Lyngby,\n",
      "Denmark ®Department of Pediatrics, University of California, San Diego, La Jolla, CA, USA\n",
      "\n",
      "Absiract\n",
      "\n",
      "Most therapeutic monoclonal antibodies in biopharmaceutical processes are produced in Chinese\n",
      "hamster ovary (CHO) cells. Technological advances have rendered the selection procedure for\n",
      "higher producers a robust protocol. However, information on molecular mechanisms that impart\n",
      "the property of hyper-productivity in the final selected clones is currently lacking. In this study, an\n",
      "IgG-producing industrial cell line and its methotrexate (MTX)-amplified progeny cell line were\n",
      "analyzed using transcriptomic, proteomic, phosphoproteomic, and chromatin immunoprecipitation\n",
      "(ChIP) techniques. Computational prediction of transcription factor binding to the transgene\n",
      "cytomegalovirus (CMV) promoter by the Transcription Element Search System and upstream\n",
      "regulator analysis of the differential transcriptomic data suggested increased in vivo CMV\n",
      "promoter-cAMP response element binding protein (CREB 1) interaction in the higher producing\n",
      "cell line. Differential nuclear proteomic analysis detected 1.3-fold less CREB1 in the nucleus of\n",
      "the high productivity cell line compared with the parental cell line. However, the differential\n",
      "abundance of multiple CREB1 phosphopeptides suggested an increase in CREB1 activity in the\n",
      "higher producing cell line, which was confirmed by increased association of the CMV promotor\n",
      "with CREB1 in the high producer cell line. Thus, we show here that the nuclear proteome and\n",
      "phosphoproteome have an important role in regulating final productivity of recombinant proteins\n",
      "from CHO cells, and that CREB1 may play a role in transcriptional enhancement. Moreover,\n",
      "CREB 1 phosphosites may be potential targets for cell engineering for increased productivity.\n",
      "\n",
      "Published by Elsevier B.V. This is an open access article under the CC BY license (http://creativecommons.org/licenses/by/4.0/)\n",
      "\n",
      "“Corresponding author. ssharfstein@ sunypoly.edu. (S.T. Sharfstein).\n",
      "These two authors contributed equally to this work.\n",
      "2Current Address: Delaware Biotechnology Institute, University of Delaware, Newark, DE, 19711.\n",
      "\n",
      "Supplementary data to this article can be found online at https://doi.org/10.1016/j.crbiot.2019.09.001,\n",
      "\n",
      "Declaration of competing interest\n",
      "The authors declare no conflict of interest.\n",
      " duosnueyy Jouiny yduosnueyy Jouny duosnueyy Jouiny\n",
      "\n",
      "duosnueyy soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "histone deacetylase 1. Mol. Cell. Biol 26, 7913-7928. 10.1128/MCB.01220-06. [PubMed:\n",
      "16940178]\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      "\n",
      "Page 17\n",
      " duosnueyy Jouiny yduosnueyy Jouny diuosnueyy Joujny\n",
      "\n",
      "diuosnuey souny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 3\n",
      "\n",
      "transgene copy number but also by transcriptional enhancement in higher producer CHO cell\n",
      "lines (Jiang et al., 2006). Thus, in these clones, the process of transcription is the rate-\n",
      "limiting step in recombinant antibody production. Further work demonstrated that treatment\n",
      "with sodium butyrate can improve gene expression in these clones (Jiang and Sharfstein,\n",
      "2008). Sodium butyrate is a known histone deacetylase inhibitor (Yin et al., 2018) and\n",
      "improved productivity may occur by increasing the accessibility of transgene to the\n",
      "transcriptional machinery. Therefore, productivity can potentially be improved by altering\n",
      "the DNA-protein interactions in the cells.\n",
      "\n",
      "The molecular basis for maximal expression from a defined section of DNA is dependent on\n",
      "the state of the chromatin. Changes in gene expression are governed by factors outside the\n",
      "realm of sequence information (Dahodwala and Sharfstein, 2014). These epigenetic changes\n",
      "are cell-type specific (Feichtinger et al., 2016; Akopov et al., 2006). Based upon epigenetic\n",
      "mechanisms, many strategies have been devised both to generate stably transfected clones as\n",
      "well as to increase specific productivity (Dahodwala and Sharfstein, 2017). While there has\n",
      "been considerable success in exploiting these observations to improve specific productivity,\n",
      "there is no clear understanding of the role of the transcriptional proteins involved. Recent\n",
      "computational and experimental studies exploring the interactions of transcription factors\n",
      "with the cytomegalovirus (CMV) promoter in the context of transient transfection and\n",
      "production of the reporter proteins secreted alkaline phosphatase and green fluorescent\n",
      "protein identified several transcription factor regulatory elements in the CMV promoter that\n",
      "affected transcription, particularly the cAMP response element (CRE) and nuclear factor\n",
      "kappa-light-chain-enhancer of activated B cells (NF«B) as positive regulatory elements and\n",
      "the binding site for the zinc finger regulatory protein YY1 as a negative regulatory site\n",
      "(Brown et al., 2015; Brown et al., 2014).\n",
      "\n",
      "While these previous studies provide insight into potential interactions of transcription\n",
      "factors with the CMV promoter in CHO cells, they do not address the potential changes in\n",
      "transcription factor regulation during cell line selection for stable clones or chromatin\n",
      "modification during stable incorporation of transgenes into the host chromosomes. For\n",
      "example, we previously observed variability in metabolic behavior between different clones,\n",
      "all from the same host cell line, producing the same recombinant product, grown in the same\n",
      "medium under the same culture conditions, presumably as a result of modifications that\n",
      "occurred during the selection process (Dahodwala et al., 2012). Whether this occurs as a\n",
      "result of stress from increased productivity, increasing levels of MTX or other factors\n",
      "remains to be elucidated. In the mutable genome of the CHO cells, the changes in chromatin\n",
      "and nuclear proteome resulting from such adaptations will have a profound effect on the\n",
      "mechanism of productivity of the derived clones. Changes in protein expression and post-\n",
      "translational changes such as phosphorylation lead to nuclear translocation of transcription\n",
      "factors and subsequent changes in DNA binding (Kaushik et al., 2018). Many cofactors\n",
      "themselves may exhibit histone acetylation activities, thereby modifying the chromatin\n",
      "accessibility and subsequent gene regulation (Zupkovitz et al., 2006). These observations\n",
      "indicate that transgene expression may be affected by inherent differences in levels and\n",
      "modifications of transcription-factor binding proteins and their subsequent interaction with\n",
      "the promoters in different cell lines. In this study, comparative phosphoproteomic data were\n",
      "gathered from a mAb-producing clone (AO) and its MTX-amplified progeny (A1), using\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " ydiosnueyy soulny ydiosnueyy Joulny yduosnueyy soulny\n",
      "\n",
      "yduosnueyy soulny\n",
      "\n",
      "Dahodwala et al. Page 2\n",
      "\n",
      "Keywords\n",
      "\n",
      "Chromatin immunoprecipitation (ChIP); CHO cell line selection; Nuclear proteomics;\n",
      "Transcriptional regulation\n",
      "\n",
      "1. Introduction\n",
      "\n",
      "The development of hybridoma technology by Kohler and Milstein (KGhler and Milstein,\n",
      "1975), set the stage for the development of monoclonal antibodies (mAbs) as tools in\n",
      "research, diagnostic agents, and revolutionary therapeutic agents, treating a wide range of\n",
      "indications. Chinese hamster ovary (CHO) cells have emerged as the dominant host for\n",
      "production of protein biopharmaceuticals, particularly monoclonal antibodies. As increasing\n",
      "numbers of therapeutic protein candidates enter various stages of development,\n",
      "biopharmaceutical companies are seeking innovative solutions to deliver this pipeline\n",
      "(Tejwani et al., 2018). Therefore, in this competitive market, it is essential to reduce time to\n",
      "market while maintaining desired quality attributes. Moreover, due to the large doses of\n",
      "antibody therapeutics required over an extended period for some indications, manufacturing\n",
      "capacity becomes an issue. To meet the high demand for biopharmaceuticals, many\n",
      "companies have built large-scale manufacturing plants containing multiple 10,000 L or\n",
      "larger cell-culture bioreactors. In this paradigm, high-producing cell lines significantly\n",
      "impact the development timelines and reduce costs by reducing needed bioreactor capacity\n",
      "and process cycles. Traditional cloning methods for cell line production and selection have\n",
      "many shortcomings and are labor intensive and time consuming. Even with the development\n",
      "of platform technologies and processes, each biopharmaceutical molecule still requires\n",
      "labor-intensive clone selection. Currently, there is a lack of understanding of the cellular\n",
      "organization and mechanism of high productivity, hindering the rapid development and\n",
      "selection of higher producing clones.\n",
      "\n",
      "Cell line development is currently performed in the following steps:\n",
      "\n",
      "1. A host cell line is transfected with a transgene-harboring plasmid via an\n",
      "optimized protocol. Flow cytometry-based staining techniques are frequently\n",
      "used to confirm transfected clones.\n",
      "\n",
      "2. Pools are amplified and selected using a chemical reagent.\n",
      "3. Single cells are isolated, scaled up, and adapted.\n",
      "4. A final clone is selected based on titer and stability.\n",
      "\n",
      "A top clone is not merely isolated from a pool of differential producers, rather the cell line\n",
      "adaptation to screening pressures results in genomic and phenotypic changes that gives rise\n",
      "to the final top clone (Noh et al., 2018). Methotrexate (MTX) amplification is routinely\n",
      "employed in dihydrofolate reductase-negative (DHFR7) systems to select for higher\n",
      "producers, with similar amplification performed using methionine sulfoximine (MSX) in\n",
      "glutamine synthetase deficient (GS~) systems. We previously characterized various CHO\n",
      "cell clones producing the same recombinant humanized monoclonal antibody and observed\n",
      "that MTX amplification leads to increased productivity by not only causing an increase in\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Joulny yduosnueyy Joulny duosnueyy Jouiny\n",
      "\n",
      "yduosnueyy souny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 16\n",
      "\n",
      "retroviral elements. J. Proteome Res 18, 2433-2445. 10.1021/acs.jproteome.8b00935. [PubMed:\n",
      "31020842]\n",
      "\n",
      "Love MI, Huber W, Anders S, 2014 Moderated estimation of fold change and dispersion for RNA-seq\n",
      "data with DESeq2. Genome Biol. 15, 550 10.1186/s13059-014-0550-8. [PubMed: 25516281]\n",
      "\n",
      "Noh SM, Shin S, Lee GM, 2018 Comprehensive characterization of glutamine synthetase-mediated\n",
      "selection for the establishment of recombinant CHO cells producing monoclonal antibodies. Sci.\n",
      "Rep 10.1038/s41598-018-23720-9.\n",
      "\n",
      "Ortega-Martinez S, 2015 A new perspective on the role of the CREB family of transcription factors in\n",
      "memory consolidation via adult hippocampal neurogenesis. Front. Mol. Neurosci 8, 46 10.3389/\n",
      "fnmol.2015.00046. [PubMed: 26379491]\n",
      "\n",
      "Potel CM, Lemeer S, Heck AJR, 2019 Phosphopeptide fragmentation and site localization by mass\n",
      "spectrometry: an update. Anal. Chem 91, 126-141. 10.1021/acs.analchem.8b04746. [PubMed:\n",
      "30457327]\n",
      "\n",
      "Qin JY, Zhang L, Clift KL, Hulur I, Xiang AP, Ren B-Z, Lahn BT, 2010 Systematic comparison of\n",
      "constitutive promoters and the doxycycline-inducible promoter. PLoS One 5, e10611 10.1371/\n",
      "journal.pone.0010611. [PubMed: 20485554]\n",
      "\n",
      "Rao X, Huang X, Zhou Z, Lin X, 2013 An improvement of the 2“(-delta delta CT) method for\n",
      "quantitative real-time polymerase chain reaction data analysis. Biostat. Bioinforma. Biomath 3,\n",
      "71-85. http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4280562/. [PubMed: 25558171]\n",
      "\n",
      "Sakamoto K, Huang B-W, Iwasaki K, Hailemariam K, Ninomiya-Tsuji J, Tsuji Y, 2010 Regulation of\n",
      "genotoxic stress response by homeodomain-interacting protein kinase 2 through phosphorylation\n",
      "of cyclic AMP response element-binding protein at serine 271. Mol. Biol. Cell 10.1091/\n",
      "mbc.e10-01-0015.\n",
      "\n",
      "Schug J, 2008 Using TESS to predict transcription factor binding sites in DNA sequence. Curr. Protoc.\n",
      "Bioinformatics 10.1002/0471250953.bi0206s21 Chapter 2. Unit 2.6.\n",
      "\n",
      "Shanware NP, Zhan L, Hutchinson J. a, Kim SH, Williams LM, Tibbetts RS, 2010 Conserved and\n",
      "distinct modes of CREB/ATF transcription factor regulation by PP2A/ B56gamma and genotoxic\n",
      "stress. PLoS One 5, e12173 10.1371/journal.pone.0012173. [PubMed: 20730097]\n",
      "\n",
      "Tejwani V, Andersen MR, Nam JH, Sharfstein ST, 2018 Glycoengineering in CHO cells: advances in\n",
      "systems biology. Biotechnol. J 13, 1700234 10.1002/biot.201700234.\n",
      "\n",
      "Trinh AT, Kim SH, Chang HY, Mastrocola AS, Tibbetts RS, 2013 Cyclin-dependent kinase 1-\n",
      "dependent phosphorylation of cAMP response element-binding protein decreases chromatin\n",
      "occupancy. J. Biol. Chem 10.1074/jbce.M113.464057.\n",
      "\n",
      "Veith N, Ziehr H, MacLeod RAF, Reamon-Buettner SM, 2016 Mechanisms underlying epigenetic and\n",
      "transcriptional heterogeneity in Chinese hamster ovary (CHO) cell lines. BMC Biotechnol. 16, 6\n",
      "10.1186/s12896-016-0238-0. [PubMed: 26800878]\n",
      "\n",
      "Villén J, Gygi SP, 2008 The SCX/IMAC enrichment approach for global phosphorylation analysis by\n",
      "mass spectrometry. Nat. Protoc 3, 1630-1638. 10.1038/nprot.2008.150. [PubMed: 18833199]\n",
      "\n",
      "Wang J, Mauvoisin D, Martin E, Atger F, Galindo AN, Dayon L, Sizzano F, Palini A, Kussmann M,\n",
      "Waridel P, Quadroni M, Dulié V, Naef F, Gachon F, 2017 Nuclear proteomics uncovers diurnal\n",
      "regulatory landscapes in mouse liver. Cell Metab. 10.1016/j.cmet.2016. 10.003.\n",
      "\n",
      "Wu X, McMurray CT, 2001 Calmodulin kinase II attenuation of gene transcription by preventing\n",
      "cAMP response element-binding protein (CREB) dimerization and binding of the creb-binding\n",
      "protein. J. Biol. Chem 10.1074/jbc.M006727200.\n",
      "\n",
      "Xu X, Nagarajan H, Lewis NE, Pan S, Cai Z, Liu X, Chen W, Xie M, Wang W, Hammond S, Andersen\n",
      "MR, Neff N, Passarelli B, Koh W, Fan HC, Wang JJJ, Gui Y, Lee KH, Betenbaugh MJ, Quake SR,\n",
      "Famili I, Palsson BO, Wang JJJ, 2011 The genomic sequence of the Chinese hamster ovary\n",
      "(CHO)-K1 cell line. Nat. Biotechnol 29, 735-741. 10.1038/nbt.1932. [PubMed: 21804562]\n",
      "\n",
      "Yin B, Wang Q, Chung C-Y, Ren X, Bhattacharya R, Yarema KJ, Betenbaugh MJ, 2018 Butyrated\n",
      "ManNaAc analog improves protein expression in Chinese hamster ovary cells. Biotechnol. Bioeng\n",
      "115, 1531-1541. 10.1002/bit.26560. [PubMed: 29427449]\n",
      "\n",
      "Zupkovitz G, Tischler J, Posch M, Sadzak I, Ramsauer K, Egger G, Grausenburger R, Schweifer N,\n",
      "Chiocca S, Decker T, Seiser C, 2006 Negative and positive regulation of gene expression by mouse\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Jouiny yduosnueyy Jouny duosnueyy Joujny\n",
      "\n",
      "jdiosnuey Joulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 12\n",
      "\n",
      "CpG island and is prone to silencing by methylation. Removing CpG islands from the\n",
      "promoter tends to mitigate gene silencing but does not prevent changes in expression due to\n",
      "histone modifications (Ho et al., 2016). While changes in transcription can occur from\n",
      "differences in histone modifications, in the present study we found fairly limited changes in\n",
      "the histone deacetylase (HDAC) and related protein mRNA levels (Supplementary Fig. S1)\n",
      "and even fewer changes at the proteomic and phosphoproteomic levels (Supplementary\n",
      "Table 2) in the clones under study, leading us to focus on other proteomic and\n",
      "phosphoproteomic differences in the amplified clones.\n",
      "\n",
      "CREB | is a transcriptional activator that undergoes complex phosphoregulation by multiple\n",
      "kinases at multiple phosphosites. Ser-133 is the most studied phosphosite on CREB 1.\n",
      "Phosphorylation of Ser-133 by PKA activates CREB1 by increasing its DNA binding\n",
      "activity, promoting the recruitment of the co-activator proteins CBP and p300. Although we\n",
      "did not find differential phosphorylation of Ser-133 in our phosphoproteomic data, three\n",
      "other phosphopeptides containing four phosphosites were identified as differentially\n",
      "phosphorylated. Various studies have shown phosphosites other than Ser-133 also play a\n",
      "crucial role in CREB 1 activity. Hence, it is important to discuss the phosphorylation pattern\n",
      "of these phosphosites to understand the complex mechanism of CREB1 regulation by\n",
      "phosphorylation. For example, we found that phosphopeptide ILNDLsSDAPGVPR\n",
      "(Ser-142) exhibited 1.52-fold higher phosphorylation in the low producer AO cell line.\n",
      "Phosphorylation of Ser-142 by CamKII inhibits CREB 1 transactivation in CV 1 mammalian\n",
      "cells by preventing CREB1 dimerization (Wu and McMurray, 2001). Furthermore, the\n",
      "differentially abundant phosphopeptide RLFSGTQISTIAESEDsQEs VDS VTDSQK\n",
      "(Ser-111 and, Ser-114) contains two phosphosites associated with the ataxia-telangiectasia-\n",
      "\n",
      "mutated and casein kinase 1 (ATM/CK) cluster, a group of closely spaced and conserved\n",
      "serine phosphosites at the N-terminal portion of CREB1 (S108, S111, $114, S117 and\n",
      "$121). The phosphorylation of these phosphosites by ATM/CK in response to DNA damage\n",
      "inhibits CREB1 mediated transcription. However, ATM-independent phosphorylation of the\n",
      "ATM/CK cluster positively regulates CREB |-mediated transcription by promoting nuclear\n",
      "translocation of cAMP-regulated transcriptional coactivators (Kim et al., 2016). Also, in\n",
      "response to genotoxic stress, phosphorylation of Ser-111 of CREB1 by ATM primes\n",
      "phosphorylation of Ser-108, Ser-114, and Ser-117 by CK1 and CK2, ultimately leading to\n",
      "inhibitory phosphorylation of Ser-121 on CREB1 by ATM. However, DNA-damage-\n",
      "independent phosphorylation of the ATK/CK cluster does not promote Ser-121\n",
      "phosphorylation (Shanware et al., 2010). In our data, we did not find phosphorylation of\n",
      "Ser-121, suggesting potential genotoxic stress-independent phosphorylation of the ATM/CK\n",
      "cluster on CREB 1 in the high productivity cell line, Al. Genotoxic stress also promotes\n",
      "inhibitory phosphorylation of Ser-271 on CREB1 by homeodomain-interacting protein\n",
      "kinase 2 (HIPK2) (Trinh et al., 2013). We found twofold increased abundance of\n",
      "phosphopeptide TAPTSTIAPGV VMAs SPALPTQPAEEAAR (Ser-271) in the high-\n",
      "producer A1 cell line. Paradoxically, studies have shown an increase in transactivation\n",
      "activity of CREB1 in response to Ser-271 phosphorylation via recruitment of CBP and p300\n",
      "(Sakamoto et al., 2010). Furthermore, homeodomain interacting protein kinase 2 (HIPK2)\n",
      "has been shown to induce phosphorylation of Ser-2361, Ser-2363, Ser-2371, Ser-2376, and\n",
      "Ser-2381 residues on CBP. It has been suggested, however, that HIPK2 enhances the\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Jouiny yduosnueyy Jouny duosnueyy Jouny\n",
      "\n",
      "yduosnuey soulny\n",
      "\n",
      "Page 6\n",
      "\n",
      "2.6. Nuclear proteomics and phosphoproteomics\n",
      "\n",
      "For proteomic analysis, cells from three biological replicates per condition were harvested at\n",
      "the mid-exponential phase of the culture. The nuclear proteomic fractions were enriched\n",
      "using NE-PER Nuclear and Cytoplasmic Extraction Reagents (Thermo Scientific — 78,833)\n",
      "as per the manufacturer’s guidelines. Protein quantification was carried out using Quick\n",
      "Start Bradford protein Assay (BioRad). To prepare the samples for mass spectrometry\n",
      "analysis, 1 mg of protein lysate from each sample was reduced by adding dithiothreitol to a\n",
      "final concentration of 5 mM and incubated at 56 °C for 25 min. Samples were then alkylated\n",
      "by adding iodoacetamide to a final concentration of 14 mM and incubated for 30 min at\n",
      "room temperature in the dark. Alkylated samples were then vortexed and diluted at a ratio of\n",
      "1:5 in 25 mM Tris-HCl. Protein samples were subsequently digested using trypsin (MS\n",
      "grade, Thermo Fisher Scientific) at 1:50 enzyme: substrate ratio. After a 4 h initial\n",
      "incubation at 37 °C, a further addition of trypsin at 1:100 enzyme: substrate ratio was\n",
      "performed followed by overnight incubation. After overnight digestion, trifluoroacetic acid\n",
      "(TFA) was added to each sample to a final concentration of 0.4% to inactivate trypsin.\n",
      "Peptides from the digested protein were concentrated and desalted using Sep-Pak C-18\n",
      "columns with negative pressure (Villén and Gygi, 2008). Ten percent of the eluate was\n",
      "aliquoted for total proteome analysis. The remaining 90% was used for phosphopeptide\n",
      "enrichment using Fe-NTA (IMAC) spin columns (Pierce, Thermo Fisher Scientific) as per\n",
      "manufacturer’s instructions. Non-enriched peptide and phosphopeptide sample\n",
      "concentrations were determined using a Nanodrop One (Laptech International, UK).\n",
      "\n",
      "2.7. LC-MS/MS analysis\n",
      "\n",
      "Both enriched phosphopeptide samples and peptides previously collected for total proteomic\n",
      "analysis were dried in a SpeedVac vacuum concentrator and resuspended in 0.1% formic\n",
      "acid (FA) containing 2% acetonitrile (ACN). Peptide volume equivalent to | yg total protein\n",
      "was injected by autosampler for LC-MS/MS analysis using an UltiMate 3000 nanoRSLC\n",
      "system (Thermo Scientific) coupled in-line with an Orbitrap Fusion Tribrid mass\n",
      "spectrometer (Thermo Scientific). Prior to the nanoLC separation, samples were first loaded\n",
      "onto the trapping column (PepMap100, C18, 300 um x 5 mm) for 3 min at a flow rate of 25\n",
      "uL/min with 2% (v/v) ACN, 0.1% (v/v) TFA. The trapped peptides were back-flushed onto\n",
      "the analytical column (Easy-Spray C18 75 um x 250 mm, 2 pm bead diameter column)\n",
      "using a gradient of 98% A (0.1% (v/v) FA): 2% B (80% (v/v) ACN, 0.08% (v/v) FA) to 35%\n",
      "B over 120 min at a flow rate of 300 nL/min.\n",
      "\n",
      "Data-dependent product ion mode was applied for both non-enriched and phosphopeptide-\n",
      "enriched MS analysis. For peptide precursor fragmentation and detection, the full MS survey\n",
      "scan (m/z 380-1500) was performed at a resolution of 120,000 with the automatic gain\n",
      "control (AGC) target set to 5 x 10°. Peptides with charge states between 2 and 7 were\n",
      "selected for MS/MS with the instrument running in top speed mode with a cycle time of 3 s.\n",
      "Dynamic exclusion was enabled with the repeat count set to 1, exclusion duration set to 60 s\n",
      "and a mass tolerance of +/— 10 ppm.\n",
      "\n",
      "For non-enriched peptide samples, MS? was performed following quadrupole isolation with\n",
      "HCD fragmentation using normalized collision energy of 28% in the ion trap (IT). MS?\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " yduosnueyy Jouiny yduosnueyy Jouny \\duosnueyy Jouny\n",
      "\n",
      "duosnuey soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 7\n",
      "\n",
      "spectra were acquired with a fixed first m/z of 100 and an intensity threshold of 5000. AGC\n",
      "was set to accumulate 1 x 104 ions and the maximum injection time was 35 ms.\n",
      "\n",
      "For phosphopeptide-enriched peptide samples, multistage activation (MSA) was performed\n",
      "following quadrupole isolation for CID fragmentation with the normalized collision energy\n",
      "set to 32%, CID activation time of 10 ms and activation Q of 0.25 in the IT. An intensity\n",
      "threshold of 10,000 was used. The neutral loss mass for MSA was 97.9673, AGC was set to\n",
      "accumulate 2 x 10+ ions and the maximum injection time was 90 ms.\n",
      "\n",
      "28, Quenititetive lesel-iree WCWSMS anehysis\n",
      "\n",
      "Relative quantitative label-free LC-MS analysis of the total proteome and phosphoproteome\n",
      "fractions from the two cell lines was carried out using Progenesis QI for Proteomics\n",
      "(Nonlinear Dynamics, Waters) in conjunction with Proteome Discoverer 2.2 (Thermo\n",
      "Scientific) for protein identification utilizing Sequest HT (Eng et al., 1994) search algorithm\n",
      "as previously described (Henry et al., 2017). Raw files generated from the MS/MS analysis\n",
      "were imported into Progenesis QI, and automatic reference alignment was carried out to\n",
      "account for retention time variability between LC runs. Upon alignment of all runs,\n",
      "identified features were filtered based on ANOVA p-value <0.05 between experimental\n",
      "groups. For proteomic and phosphoproteomic analysis, a Mascot generic file (mgf) was\n",
      "generated from all exported MS/MS spectra and analyzed using Proteome Discoverer v.2.2\n",
      "(Thermo Fisher Scientific) in conjunction with SEQUEST. Peak lists were searched against\n",
      "a proteogenomic draft annotation for the newly assembled Chinese hamster genome which is\n",
      "experimentally annotated using RNA-Seq, proteomics, and Ribo-Seq (Li et al., 2019).\n",
      "Database search parameters were set to allow MS1 tolerance of 10 ppm; MS? mass tolerance\n",
      "of 0.6 Da for ion trap detection; enzyme specificity was set as trypsin with two missed\n",
      "cleavages allowed; carbamidomethylation of cysteine was set as a fixed modification; and\n",
      "phosphorylation of serine, threonine, and tyrosine and oxidation of methionine (for\n",
      "phosphopeptide analysis) were set as variable modifications. For phosphosite identification,\n",
      "the PhosphoRS algorithm (Potel et al., 2019) was run through ProteomeDiscover 2.2 using\n",
      "diagnostic fragment ions and analyzer-specific fragment ion tolerances, as described above.\n",
      "Data were filtered to a 1% false discovery rate (FDR) on PSMs using automatic decoy\n",
      "searching in SEQUEST and by applying a phosphosite probability score of 75% or greater\n",
      "for S, T or Y amino acids in PhosphoRS (Potel et al., 2019). A statistical criterion of\n",
      "ANOVA p-value ‘0.05 and fold change cut-off 21.5 at the protein level was applied between\n",
      "experimental groups. Proteins with two or more unique peptides and phosphoproteins with\n",
      "any unique phosphopeptides that passed these criteria were then deemed to be differentially\n",
      "expressed between the relevant experimental groups being analyzed.\n",
      "\n",
      "2.9. Purification of the DNA-protein complex\n",
      "\n",
      "Chromatin immunoprecipitation was performed using the ChIP-IT kit (53008 Active Motif,\n",
      "Carlsbad CA) according to the manufacturer’s instructions. Briefly, 4 x 107 cells from each\n",
      "cell line were harvested at day 3 and incubated with 30 mL fresh medium containing 1.5 mL\n",
      "36% formaldehyde (47630 Sigma-Aldrich, St Louis, MO) for 10 min to crosslink the DNA-\n",
      "associated proteins to the chromatin. The reaction was stopped by washing the cells with\n",
      "phosphate buffered saline (PBS) and incubating with Glycine Stop-fix solution for 10 min.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " ydiosnueyy JouIny ydiosnueyy Joylny yduosnueyy soulny\n",
      "\n",
      "yduosnueyy soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 13\n",
      "\n",
      "transcriptional activity of CBP by antagonizing the repressive action of cell cycle regulatory\n",
      "domain 1 (CRD1), located between amino acids 977 and 1076 in HPIK2 independent\n",
      "manner (Kovacs et al., 2015). We observed the differential abundance of CBP C-terminal\n",
      "phosphopeptide EEEESSANGTASQstsPSQPR (1061-1083 region) on Ser-1074, Thr-1075\n",
      "and Ser-1076. The phosphorylation status of these residues has not been investigated for\n",
      "their implication in CBP transactivation activity.\n",
      "\n",
      "In addition to protein CREB1, the cAMP response element binding protein family also\n",
      "contains transcription factors CREM and ATF, which share a high degree of sequence and\n",
      "structural similarity. As described above, we observed differential abundance of the\n",
      "phosphopeptide containing Ser-271 on CREB]; interestingly, we also found the\n",
      "corresponding peptide of CREM harboring phosphosites Ser-271, Ser-274, Ser-277 and\n",
      "Ser-286 with 1.7-fold higher phosphorylation in high producer A1 cells. Despite being a\n",
      "close member of the CREB family and previously implicated in transcription regulation,\n",
      "CREM and ATF! transcription factors have not been subjected to the same scientific inquiry\n",
      "as CREB1.\n",
      "\n",
      "5. Concluding remarks\n",
      "\n",
      "Transcription initiation occurs when the transcriptional machinery binds to the promoter\n",
      "regions of genes. Subsequently, MRNA synthesis occurs and then proteins are trafficked,\n",
      "folded and secreted. Recombinant gene expression in biopharmaceutical processes is\n",
      "frequently driven by the CMV promoter. Many studies have demonstrated the importance of\n",
      "employing vector design, UCOE, and MARs elements to influence the chromatin state to\n",
      "augment the expression of therapeutic proteins in mammalian cells (Veith et al., 2016).\n",
      "However, there is little evidence of the role of the nuclear proteome and the interaction of\n",
      "DNA promoters with transcription factors to demonstrate the mechanism of transcriptional\n",
      "regulation. In this study, two cell clones exhibiting different productivity levels were\n",
      "investigated to determine the mechanism behind higher productivity. From a previous\n",
      "characterization, higher productivity in the selected clones was determined to be influenced\n",
      "by the increased transcription within the higher producer cell lines.\n",
      "\n",
      "Based upon the binding consensus of the transcription factors in the database, we\n",
      "hypothesized that CREB 1 has a large likelihood of binding to the promoter region. By\n",
      "employing ChIP, we were able to study the interactions with the chromatin in each of the\n",
      "cell lines investigated. Our results indicate ~6-fold increased binding in the higher producer\n",
      "\n",
      "cell line. CREB 1 is also implicated in other aspects of gene regulation, including histone\n",
      "modifications by CBP, a histone acetyltransferase, (Khan et al., 2017) (Levine et al., 2005)\n",
      "and in interactions with the DNA methyltransferases Dnmt3a and Dnmt3b (Hervouet et al.,\n",
      "2009). As the consensus binding region of the transcriptional factors falls on the site of a\n",
      "CpG island within the CMV promoter, we expect a complex interplay in the function of\n",
      "transcription factor with the DNA in cells adapted to very high productivity phenotypes. Our\n",
      "understanding of the role of CHO nuclear proteome needs to evolve for us to understand the\n",
      "complex interplay between the transcriptional machinery and the high productivity\n",
      "phenotype.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Joulny yduosnueyy Joulny duosnueyy Jouyny\n",
      "\n",
      "\\duosnueyy souny\n",
      "\n",
      "Dahodwala et al. Page 5\n",
      "\n",
      "2.1.3. Antibody assay—Antibody titers were determined by ELISA using a Human\n",
      "IgG ELISA Antibody Pair Kit (Stemcell Technologies) as per the manufacturer’ s\n",
      "instructions.\n",
      "\n",
      "2.2. Antibodies\n",
      "\n",
      "Antibodies to RNA polymerase II were provided with the ChIP IT kit (Active Motif,\n",
      "Carlsbad CA). Antibodies to CREB1 (39013), NFxB (40916) and Sp1 (39058) were\n",
      "purchased separately from Active Motif, Carlsbad, CA.\n",
      "\n",
      "2.3. Prediction of transcriptional proteins interacting with CMV promoter region\n",
      "\n",
      "The Transcription Element Search System (TESS) previously available at http://\n",
      "www.cbil.upenn.edu/cgi-bin/tess/tess is a database that contains the various binding\n",
      "consensus sequences that are recorded by experimental investigation of the transcriptional\n",
      "proteins (Schug, 2008). It can identify binding sites using site or consensus strings and\n",
      "positional weight matrices from the TRANSFAC, JASPAR, IMD, and the CBIL-GibbsMat\n",
      "database. By querying the database with the CMV promoter sequence, we were able to\n",
      "generate a probability score of CMV promoter-region interactions with all transcription\n",
      "factor proteins in the database.\n",
      "\n",
      "2.4. RNA-seq analysis\n",
      "\n",
      "RNA-Seq data were generated as previously reported (Chiang et al., 2019). Briefly, total\n",
      "RNA was isolated using a Qiagen RNeasy Plus Mini Kit as per the manufacturer’s\n",
      "instructions. RNA quality was verified using an Agilent Bioanalyzer prior to library\n",
      "preparation. Library preparation was performed with an Illumina TruSeq Stranded mRNA\n",
      "Library Prep Kit High Throughput (Catalog ID: RS-122—2103), according to manufacturer’s\n",
      "protocol. Final RNA libraries were first quantified by Qubit HS and then QC on Fragment\n",
      "Analyzer (from Advanced Analytical). The final pool of libraries was analyzed on the\n",
      "Illumina NextSeq platform with high output flow cell configuration (NextSeq® 500/550\n",
      "High Output Kit v2 (300 cycles) FC-404—2004).\n",
      "\n",
      "2.5. RNA-Seq data processing\n",
      "\n",
      "The RNA libraries were mapped to the CHO genome (C_griseus_v1.0) (Lewis et al., 2013;\n",
      "Xu et al., 2011) using STAR aligner (v. 2.5.4b) (Dobin et al., 2013). Alignments were\n",
      "processed to quantify gene expression counts with HTSeq-count (v. 0.7.2) (Anders et al.,\n",
      "2015). Genes with very low expression (less than one count in at least two samples) or of\n",
      "zero variance were excluded from further downstream analysis. DESeq2 with default\n",
      "parameters (Love et al., 2014) was used to estimate the differential expression between the\n",
      "Al and AO samples, with a positive fold change denoting higher expression in Al. The raw\n",
      "sequencing files and count matrix were deposited to SRA and GEO (accession number\n",
      "GSE133511). To comply with intellectual property requirements, the sequencing data were\n",
      "processed to exclude unmapped reads. This results in <5% reduction in available reads.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Jouiny yduosnueyy Jouny duosnueyy Jouiny\n",
      "\n",
      "yduosnueyy soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 11\n",
      "\n",
      "immunoprecipitation was carried out, employing antibodies to these transcriptional proteins\n",
      "as well as to Sp1, a negative control.\n",
      "\n",
      "Immunoprecipitated chromatin was purified and then subjected to RT-gPCR to quantify the\n",
      "number of copies of GAPH and CMV promoter regions bound to each transcription factor.\n",
      "After immunoprecipitation, PCR-derived Ct values were normalized to the respective input\n",
      "DNA values to serve as a loading control. We observed a ~6-fold difference in the CREB 1\n",
      "association with the CMV promoter chromatin in the amplified progeny A1 relative to the\n",
      "AO parental cell line, indicating that the CREB1 association to the CMV promoter was\n",
      "increased even after accounting for MTX amplification and subsequent increase in CMV\n",
      "copies (Fig. 3). The CMV binding to NFxB and RNA pol II was much lower than the\n",
      "binding to CREB1. As expected, the CMV binding to Sp1 was quite low as the Sp1\n",
      "transcription factor has a predicted low likelihood of association with the CMV promoter.\n",
      "\n",
      "When the immunoprecipitated DNA was assayed with GAPDH primer probes (Fig. 4), we\n",
      "observed a ~4-fold increase in the GAPDH-CREB1 association in the amplified clone Al\n",
      "relative to the progeny clone AO. These data suggest that the transcription factor CREB 1\n",
      "shows an increased association with the genomic DNA within the Al amplified progeny\n",
      "relative to the parental AO clone, which would be consistent with activation of CREB1 in the\n",
      "amplified cell lines. Despite the predicted increase in NF«B activity from the URA, no\n",
      "increased association of NFxB or RNA pol II with either CMV or GAPDH was observed.\n",
      "\n",
      "4. Discussion\n",
      "\n",
      "4.1. Role of CREB1 in transcriptional enhancement\n",
      "\n",
      "In previous studies (Jiang et al., 2006), we observed that the Al cell line has approximately\n",
      "twice the number of copies of the transgenes but expresses 5-fold more transcripts than the\n",
      "AO parental clone. Results presented here suggest increased association of CREB1 with the\n",
      "nucleus in the high producer cell line Al compared with the parental AO clone. CREB1 is a\n",
      "well-studied transcriptional activator that promotes gene expression from a variety of\n",
      "promoters. CREB becomes activated as part of the cAMP cascade. A stimulus from the\n",
      "environment causes cAMP to activate a protein kinase that in turn translocates to the nucleus\n",
      "and phosphorylates the CREB1 protein. Activated CREB1 then binds to its DNA consensus\n",
      "sequence in many promoters in the cell and activates the transcriptional machinery. The\n",
      "consensus sequence (8 -GIGACGT[AC][AGL3 Vis present in many viral and cellular\n",
      "promoters, including the constitutive CMV promoter that is used to drive gene expression in\n",
      "the clones in this study and is widely used for recombinant protein production in the\n",
      "biopharmaceutical industry. (Ortega-Martinez, 2015). Typically, CREB1 binds DNA as a\n",
      "dimer and also acts as a coactivator of transcription by associating with UTF1 (Beausoleil et\n",
      "al., 2004). The CREB-binding protein (CBP) also exhibits histone acetyltransferase (HAT)\n",
      "activity. Hence, CREB1 is suggested to have a role in the activation of transcription by DNA\n",
      "binding as well as modulating the chromatin, specifically by acetylating histones H2B and\n",
      "H4 (Harton et al., 2001).\n",
      "\n",
      "While CMV is a strong constitutive promoter, it shows great variability in activity depending\n",
      "on the cell line used for transfection (Qin et al., 2010). The CMV promoter also has a large\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Joulny \\diuosnueyy Joulny duosnueyy Jouiny\n",
      "\n",
      "duosnuey souny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 10\n",
      "\n",
      "phosphoproteins showed reduced phosphorylation and/or decreased expression in the Al\n",
      "cell line compared to the AO parental cell line. Only proteins and phosphoproteins with fold\n",
      "change 21.5 were deemed differential. 178 proteins were common between the list of\n",
      "differential proteins and differential phosphoproteins. A complete list of all nuclear proteins,\n",
      "phosphoproteins, and phosphopeptides is presented in Supplementary Table 1.\n",
      "\n",
      "Although no differential mRNA expression for CREB1 was identified in the transcriptomic\n",
      "data, 1.3-fold lower expression of total CREB1 protein was observed in the high producer\n",
      "Al cell line compared with the AO cell line (Fig. 2A, Supplementary Table 1). In the light of\n",
      "TESS prediction and URA analysis, the observed decrease in CREB 1 total protein in the\n",
      "nucleus of the A1 cell line could be due to the increased activity of CREB1 in A1 cells,\n",
      "leading to greater CREB1-DNA association, decreasing the availability of free CREB1 in\n",
      "the nuclear proteome for proteomic detection. Moreover, phosphoproteomic comparison\n",
      "between A1 and AO cell line revealed differential abundance of three phosphopeptides of\n",
      "CREB 1 corresponding to four phosphosites, i.e. IELNDLSSDAPGVPR (Ser-148),\n",
      "RLFSGTQISTIAE SEDSQESVDSVTDSQK (Ser-111, Ser-114) and TAPTSTIAPGVVMA\n",
      "SSPALPTQPAEEAAR (Ser-271). The expression pattern of these phosphopeptides is shown\n",
      "in Fig. 2B-D (See also Supplementary Table 1).\n",
      "\n",
      "3.4. CREB1: transcription regulatory relationship\n",
      "\n",
      "The expression pattern of transcription-factor target proteins can provide insight into the\n",
      "dynamics of regulation by the transcription factor. Hence, CREB 1-interacting proteins from\n",
      "TRRUST (Transcriptional Regulatory Relationships Unraveled by Sentence-based Text\n",
      "mining) database were searched in the lists of differential genes, proteins, and\n",
      "phosphoproteins between the AO and A1 cell lines. The expression pattern of the six CREB1\n",
      "target proteins identified is shown in Table 2. Two proteins that are repressed by CREB1\n",
      "activation, HMOX1 and JUN, were downregulated at the mRNA and protein level, and\n",
      "protein JUN also showed downregulation at the phosphoprotein level in the A1 cell line.\n",
      "Both proteins showed a high likelihood of inhibition in the URA analysis. Two proteins that\n",
      "are up-regulated by CREB1, NOLC1 and NDC80, were upregulated at the total protein level\n",
      "in the A1 cell line. Protein NOLC1 was also upregulated at the phosphoprotein level\n",
      "whereas the mRNA level for NOLC1 was downregulated in the A1 cell line.\n",
      "\n",
      "3.5. Chromatin immunoprecipitation\n",
      "\n",
      "Previous work in our lab has identified transcription as the rate-limiting step in the\n",
      "production of monoclonal antibodies in both the parental cells and amplified progeny in\n",
      "these cell lines (Jiang et al., 2006). Based upon the consensus binding sequences of the\n",
      "transcriptional proteins and their occurrences along the CMV promoter, bioinformatic\n",
      "analysis of the CMV sequence indicated that CREB1 exhibits a high probability of\n",
      "influencing transcription from the CMV promoter (Table 1). The URA and nuclear\n",
      "proteomic and phosphoproteomic analysis further substantiated the potential role of CREB1.\n",
      "The URA also indicated a high likelihood of NF«B activation; however, we were unable to\n",
      "find any additional proteomic or phosphoproteomic evidence to support this activation. To\n",
      "evaluate whether high and low productivity clones exhibited differential interactions\n",
      "between the CMV promoter region and these transcription factors, chromatin\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Jouiny ydiuosnueyy Jouny yduosnueyy Jouiny\n",
      "\n",
      "diuosnueyy souny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 4\n",
      "\n",
      "quantitative, label-free LC-MS proteomic techniques to demonstrate the activation and\n",
      "increased phosphorylation of CREB1 in the amplified cell line. Further, chromatin-\n",
      "transcription factor interactions were investigated by comparing the parental clone and its\n",
      "MTX-amplified progeny via chromatin immunoprecipitation (ChIP). An increased DNA-\n",
      "protein interaction in the higher producing cell lines was observed. CREB 1 transcription\n",
      "factor showed ~6-fold increased association with the cytomegalovirus (CMV) promoter in\n",
      "higher-producing cell lines. Together, these results indicate an increasing association of\n",
      "transcriptional proteins with the DNA in the higher producing clones, reinforcing the notion\n",
      "that epigenetics and nuclear proteome interplay is an important, but poorly understood driver\n",
      "of transgene expression in mammalian cells.\n",
      "\n",
      "2. Methods and materials\n",
      "\n",
      "2.1. Cell lines\n",
      "\n",
      "Chinese hamster ovary cell lines that produce a recombinant monoclonal humanized IgG\n",
      "with different specific productivities were a generous gift from an industrial collaborator.\n",
      "These cell lines were developed by co-transfecting two plasmids, one containing IgG heavy\n",
      "chain (HC) and dihydrofolate reductase (DHFR) genes and the other containing IgG light\n",
      "chain (LC) and neomycin phosphotransferase (Neo) genes. Transfected cell lines were\n",
      "initially selected in medium containing 400 ug/mL neomycin (G418). After selection, the\n",
      "neomycin was removed, and all subsequent cultures were performed in the absence of\n",
      "neomycin. Subsequently, gene amplification was performed by stepwise selection with\n",
      "increasing MTX concentrations. For these studies, a low producing parental cell line AO and\n",
      "its amplified high-producing progeny cell line Al were chosen for investigation. These cell\n",
      "lines have been previously described (Jiang et al., 2006; Jiang and Sharfstein, 2009) After\n",
      "culture medium adaptation, cells were cultured in a nonproprietary, serum-free medium\n",
      "\n",
      "(Dahodwala et al., 2012) containing hydrolysate and 5 mg/L recombinant human insulin\n",
      "(Supplementary Table 3). Growth curves and relative specific productivities are shown in\n",
      "Supplementary Fig. $2.\n",
      "\n",
      "27.1. Gell culture cond torne—For every experimental method described, miplicate\n",
      "batch suspension cultures of all cell lines were maintained in 125 mL Erlenmeyer flasks. For\n",
      "each cell line, 0.2 x 10° cells were seeded into 25 mL of medium and cultured on an orbital\n",
      "shaker at 125 rpm, 36 °C, and 5% CO . Routine subculturing was carried out for 2 passages\n",
      "after thawing before experiments were performed.\n",
      "\n",
      "2.1.2. Sampling—Samples were taken daily from suspension cultures to determine cell\n",
      "density and viability. Cell densities and viabilities were estimated by hemacytometer counts\n",
      "(Hausser Scientific, PA) or automated counting (BioRad TC10) after diluting 1:1 with 0.4%\n",
      "trypan blue solution.\n",
      "\n",
      "Cell pellets were collected at mid-exponential stage in culture (3 days after inoculation).\n",
      "Cells were harvested by centrifuging the appropriate volume of culture suspension at 1200\n",
      "rpm for 5 min.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Jouiny yduosnueyy Joulny duosnueyy Jouiny\n",
      "\n",
      "yduosnuey souny\n",
      "\n",
      "Page 9\n",
      "\n",
      "transcription factors (ATF), cAMP response element binding protein (CREB1), and\n",
      "activating protein 2 (AP-2) as shown in Table 1.\n",
      "\n",
      "3.2. RNA-Seq - upstream regulator analysis\n",
      "\n",
      "Differential gene expression analysis between the AO and A1 cell lines was performed from\n",
      "the count data obtained from RNA-Seq analysis using DEseq2 package in R. 6272 genes\n",
      "were identified as differentially expressed between the Al and AO cell lines (pagj < 0.05)\n",
      "from which 2938 genes exhibited higher expression in the higher producer progeny A1 cell\n",
      "line, whereas 3334 genes showed lower expression in Al as compared to the parental AO\n",
      "cell line.\n",
      "\n",
      "To understand key genes that regulate productivity associated pathways, Ingenuity Upstream\n",
      "Regulator Analysis (URA) was performed on the list of differentially expressed genes. URA\n",
      "analysis provides a numerical account of predicted regulator activity as Z-score value. A\n",
      "positive Z-score > 1.5 indicates potential activation of a regulatory factor and Z-Score < -1.5\n",
      "indicates potential inhibition of the regulatory factor (Kramer et al., 2013). We found 102\n",
      "regulatory factors with Z-score > 1.5 and 56 regulatory factors with Z-score < —1.5. Among\n",
      "all predicted regulatory proteins, 28 transcription factors were predicted to be activated, and\n",
      "11 transcription factors were predicted to be inhibited in the high producer A1 cell line.\n",
      "Examining the transcription factors with a high probability of interaction with the CMV\n",
      "promoter, both CREB1 and NFxB were identified by the URA as having a high probability\n",
      "of being activated in the A1 cell line compared with the AO parental cell line. Although no\n",
      "expression change for CREB1 was observed at the mRNA level, URA prediction suggested\n",
      "strong activation of CREB1 (z-score = 2.13). In addition, the NFxB complex was identified\n",
      "as likely to be activated, with a z-score = 1.7. URA analysis also predicted potential\n",
      "activation of 11 kinases and inhibition of 6 kinases in the A1 cells, an important class of\n",
      "regulatory proteins that influence transcription factor activation. A complete list of all\n",
      "transcription factors with predicted differential activation is provided in Fig. 1.\n",
      "\n",
      "3.3. Proteomic and phosphoproteomic changes associated with high productivity\n",
      "\n",
      "Previous proteomic studies have shown that changes in protein productivity in CHO cells\n",
      "bring about systemic changes in the proteome (Hausmann et al., 2018) and post-translation\n",
      "modification of cellular proteins (Henry et al., 2017). Due to the compartmentalization of\n",
      "transcriptional machinery in the nucleus, studying the nuclear proteome can provide insight\n",
      "into changes in expression and modification of proteins that influence transgene expression.\n",
      "Furthermore, nuclear proteomic studies can provide up 60% greater coverage of key\n",
      "regulatory proteins such as transcription factors, transcription co-regulators and RNA\n",
      "processing proteins (Wang et al., 2017) than whole cell proteomic analysis.\n",
      "\n",
      "Quantitative nuclear proteomic and nuclear phosphoproteomic analysis of high producer Al\n",
      "cell line and low producer AO cell line revealed 873 differentially expressed proteins (DEP)\n",
      "and 640 differential phosphoproteins (DEpP). From the DEP, 426 proteins were elevated in\n",
      "the high producer A1 cell line and 447 proteins were more highly expressed in the low\n",
      "producer AO cells. Similarly, from the nuclear phosphoproteomic analysis, 528\n",
      "phosphoproteins showed elevated phosphorylation and/or increased expression and 112\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " ydiosnueyy soulny ydiosnueyy Joulny jduosnueyy soulny\n",
      "\n",
      "yduosnueyy soulny\n",
      "\n",
      "Dahodwala et al. Page 21\n",
      "\n",
      "tated\n",
      "ol\n",
      "\n",
      "mm AO\n",
      "Co At\n",
      "\n",
      "BS\n",
      "\n",
      "immunoprecipi\n",
      "ow\n",
      "\n",
      "N\n",
      "\n",
      "CREB1 NFxB Sp1 RNA pol Il\n",
      "\n",
      "Relative GAPDH\n",
      "\n",
      "Fig. 4.\n",
      "Normalized number of copies of GAPDH promoter region binding to transcription factors in\n",
      "Al vs. AO cell line.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " ydiosnueyy soulny ydiosnueyy Joylny idiuosnueyy Joulny\n",
      "\n",
      "yduosnueyy soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "itated\n",
      "\n",
      "immunoprecipi\n",
      "\n",
      "Relative CMV\n",
      "\n",
      "Page 20\n",
      "\n",
      "mm AO\n",
      "Co At\n",
      "\n",
      "=\n",
      "ak a Oo N ©\n",
      "\n",
      "1.0\n",
      "0.5\n",
      "0.0\n",
      "\n",
      "CREB1 NF«KB Sp1 RNA pol Il\n",
      "\n",
      "Fig. 3.\n",
      "Normalized number of copies of CMV promoter region binding to transcription factors in\n",
      "Al cell line vs. AO cell line.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Joulny \\diuosnueyy Joulny duosnueyy Jouiny\n",
      "\n",
      "duosnuey souny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Page 8\n",
      "\n",
      "A final PBS wash step was used to clean the cell pellet. A sonicator (450D Branson,\n",
      "Danbury CT) fitted with a microtip was employed to disrupt the cells and shear the DNA to\n",
      "500-1500-base pair fragments. The sonicator settings were set in accordance with the tip\n",
      "manufacturer’s instructions and kept at 40% amplitude. The shearing was verified by\n",
      "separating the sheared DNA on a 1.8% agarose gel. In subsequent steps, the Protein- DNA\n",
      "complex was immunoprecipitated using antibodies to CREB1, NFxB, Spl or RNA\n",
      "polymerase II. After treatment with Proteinase K and RNase to remove cellular proteins and\n",
      "RNA, DNA fragments were purified by using silica spin columns provided with the kit. The\n",
      "final elution volume in each fraction was 100 pL. This volume was concentrated to 20 uL by\n",
      "using a Speed Vac DNA concentrator (BC-SDNA11 Savant, GMI Inc. Ramsey, Minnesota)\n",
      "to obtain an adequate concentration of DNA template for RT-qPCR. ChIP was performed on\n",
      "three separate dates with duplicate PCR analysis for each sample.\n",
      "\n",
      "2.10. RT-qPCR\n",
      "\n",
      "Real time quantitative PCR was performed using the Roche LightCycler® 480 Real-Time\n",
      "PCR System and the LightCycler 480 Mastermix (04707494001 Roche, Indianapolis, IN).\n",
      "For quantification of CMV, the probe/primers combinations were as follows: forward\n",
      "primer: gcagagctcgtttagtgaacc; reverse primer: gaggtcaaaacagcgtggat; Universal\n",
      "ProbeLibrary probe: #80 (cat.no. 04689038001, Roche, Indianapolis, IN). For quantification\n",
      "of glyceraldehyde 3-phosphate dehydrogenase (GAPDH), the probe/primers combinations\n",
      "were as follows: forward primer: cgtattggacgcctggttac; reverse primer: ggcaacaacttccactttgc;\n",
      "Universal ProbeLibrary probe: #8 (cat.no. 04685067001, Roche, Indianapolis, IN). Reaction\n",
      "conditions were set up according to the manufacturer’s instructions. Crossing points (Ct)\n",
      "were generated from the LightCycler Software. Relative quantification of the CMV\n",
      "promoter and GAPDH bound to the transcription factors was performed using the 2/delta\n",
      "delta Ct method (Rao et al., 2013). All samples were normalized to the respective input\n",
      "DNA for the ChIP reaction (e.g. AO cell line, CMV copies in input DNA) and then to sample\n",
      "3 of the AO CREB! precipitate for CMV or GAPDH, respectively.\n",
      "\n",
      "3. Results\n",
      "\n",
      "3.1. Transcription factor-binding analysis of CMV promoter\n",
      "\n",
      "In this study, protein-DNA complex interactions were examined to understand the\n",
      "transcriptional enhancement in the MTX-amplified cell lines and further elucidate\n",
      "transcriptional regulation in high productivity clones. Transcription factors associate\n",
      "upstream of the gene of interest via specific binding motifs that interact with consensus\n",
      "sequences along the promoter region to initiate transcription.\n",
      "\n",
      "The TESS web tool was used for predicting transcription factor binding sites in the CMV\n",
      "promoter sequence. The CMV promoter is a strong viral promoter system used in transgene\n",
      "expression. It has a high level of constitutive gene expression and is efficient in a broad\n",
      "range of cell types. Putative transcription factors with a high probability of binding included\n",
      "enhancer factor C (EF-C), methylated DNA-binding protein (MDBP), activator protein 1\n",
      "(AP-1), nuclear factor kappa-light-chain-enhancer of activated B cells (NF«B), activating\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " jdiuosnueyy soujny jduosnueyy soujny jdiosnueyy soujny\n",
      "\n",
      "jdiuosnueyy soulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "Table 1\n",
      "\n",
      "Top scoring transcription factors binding sites. 5 out of 548 sites are shown. From TESS: Transcription\n",
      "\n",
      "Element Search System (http://www.cbil.upenn.edu/cgi-bin/tess/t) along with the z-score values from\n",
      "\n",
      "Ingenuity URA analysis.\n",
      "\n",
      "#~= = Transcription Factor Beg = Sns\n",
      "\n",
      "1 T00230 EF-C 292 R\n",
      "T00963 MDBP.\n",
      "\n",
      "2  TO00029 AP-1 562 OR\n",
      "TO00590 NF-xB\n",
      "\n",
      "3. TOO0S1 ATF 592 OR\n",
      "T00167 CREB\n",
      "\n",
      "4  T00952 AP-2 99 N\n",
      "\n",
      "Len\n",
      "\n",
      "25\n",
      "\n",
      "Sequence\n",
      "TATGTTCCCATAGT\n",
      "AACGCCAATAG\n",
      "\n",
      "TITGACTCACGGGG\n",
      "ATTTCCAAG\n",
      "\n",
      "CCCATTGACGTCA\n",
      "ATGGG\n",
      "\n",
      "CCCMNSSS\n",
      "\n",
      "“La\n",
      "\n",
      "50.0\n",
      "\n",
      "46.0\n",
      "\n",
      "36.0\n",
      "\n",
      "10.0\n",
      "\n",
      "PuRA Z-Score\n",
      "\n",
      "2.135\n",
      "\n",
      "a oars oe a .\n",
      "La: Log likelihood score of association of transcription factor to given consensus sequence.\n",
      "\n",
      "Page 22\n",
      "\n",
      "URA Z score: Likelihood of activation (positive Z-score) or inactivation (negative Z-score) of transcription factor based on differential expression\n",
      "\n",
      "of genes regulated by the transcription factor.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " Page 23\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "uonRAnOy cALA\n",
      "\n",
      "“OUT [[99 TV oy} UI UONR[NZaIdn yuosaidar s[joo UIeIS pue UOTe[NZaIUMOp JUosaIdaI s[[99 BBULIO “(OV/TV)ON@ 3077 se uMoyYs\n",
      "aie sanfeA oBuryo ploy ‘asequiep ArojyepnSor peuondiosuey (/\\sn.1/310 vipeduis mmmay/:sdyy) LSNAAL Woy sutajoid drysuoneyas AJoywnser | GAO.\n",
      "\n",
      "2 aqeL\n",
      "\n",
      "Author Manuscript Author Manuscript Author Manuscript Author Manuscript\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " duosnueyy Joulny ydiuosnueyy Joulny diuosnuey Jouiny\n",
      "\n",
      "duosnueyy souny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "IRF7\n",
      "STAT3\n",
      "GATA2\n",
      "IRF3\n",
      "NFATC2\n",
      "STATA\n",
      "SMAD2\n",
      "HTT\n",
      "TWIST1\n",
      "CREB1\n",
      "PPP1R13L\n",
      "WWTR1\n",
      "NFE2L1\n",
      "PTTG1\n",
      "PYCARD\n",
      "SMAD3\n",
      "FOXL2\n",
      "ZBTB20\n",
      "NCOA2\n",
      "REL\n",
      "HMGB1\n",
      "ARNT\n",
      "IRF5\n",
      "MITF\n",
      "NFkB (complex)\n",
      "RUNX2\n",
      "HIFIA\n",
      "MYB\n",
      "HNFIA\n",
      "STAT6\n",
      "TFAP2C\n",
      "IRF4\n",
      "RB1\n",
      "PDX1\n",
      "HDAC6\n",
      "ERG\n",
      "PAX6\n",
      "TRIM24\n",
      "NUPR1\n",
      "\n",
      "Transcription regulators\n",
      "\n",
      "Fig. 1.\n",
      "\n",
      "2\n",
      "\n",
      "0\n",
      "\n",
      "Activation Z-score\n",
      "\n",
      "Page 18\n",
      "\n",
      "WB activatea\n",
      "Wh onivitea\n",
      "\n",
      "Ingenuity pathway analysis-Predicted activity status of transcription regulators from\n",
      "\n",
      "upstream pathway analysis.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      " ydiosnueyy soulny ydiosnueyy Joylny duosnueyy Joulny\n",
      "\n",
      "iduosnueyy Joulny\n",
      "\n",
      "Dahodwala et al.\n",
      "\n",
      "1.5*107 8 1.5x105\n",
      "iJ\n",
      "8 z\n",
      "& Fe\n",
      "3 = qos\n",
      "F 3\n",
      "3 = 2\n",
      "c\n",
      "3 B  —-5x104\n",
      "Ss °\n",
      "2 a\n",
      "a a\n",
      "°\n",
      "z 0\n",
      "Cell line Cell line\n",
      "Phospho-CREB1 Phospho-CREB1\n",
      "S-271 Ser-142\n",
      "Fold Change = 2.0 Fold Change = 0.74\n",
      "8 5x106 8 6x105\n",
      "& c\n",
      "st no]\n",
      "4x08\n",
      "5 * = 5\n",
      "2 | 4x105\n",
      "& —-3x108 S\n",
      "3 3 xk\n",
      "ry 2*106 Gy\n",
      "3 B — 2x108\n",
      "& — 1x108 a\n",
      "8 8\n",
      "£ £\n",
      "a 0 ao 0\n",
      "© e e\n",
      "Cell line Cell line\n",
      "ig. 2,\n",
      "\n",
      "Page 19\n",
      "\n",
      "Phospho-CREB1\n",
      "Total CREB1 Ser-111, Ser-114\n",
      "Fold Change = 0.65 Fold Change = 1.58\n",
      "\n",
      "CREB 1 proteomic and phosphoproteomic expression changes in the nucleus of AO and Al\n",
      "cell lines. A-Total CREB1, B-D-phosphopeptides Ser-111 and Ser-114, Ser-272, and\n",
      "Ser-142, respectively. Fold change is the relative abundance in A1 relative to AO.\n",
      "\n",
      "Curr Res Biotechnol. Author manuscript; available in PMC 2020 June 23.\n",
      "Abstrak yang relevan dan terfilter:\n"
     ]
    }
   ],
   "source": [
    "search_terms = [\"diabetes complications management\", \"innovations in diabetes treatment\", \"advancements in diabetic wound healing\"]\n",
    "user_query = \"What are the latest innovations in diabetes complications management?\"\n",
    "folder_path = \"/Users/dwiyulianto/Downloads/AI PROJECT/Journal_HHS Public Acces _Health\"  # Ganti dengan path folder gambar\n",
    "\n",
    "# Menjalankan pipeline\n",
    "multi_agent_pipeline(search_terms, user_query, folder_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "04bf9584-2827-4d3f-b843-eb9e7a7b6e63",
   "metadata": {},
   "source": [
    "# Agent untuk Question Answering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f0a94a65-5905-4f35-97ac-e4767f1ea66f",
   "metadata": {},
   "outputs": [],
   "source": [
    "@ray.remote\n",
    "class QuestionAnsweringAgent:\n",
    "    def __init__(self):\n",
    "        self.model = T5ForConditionalGeneration.from_pretrained(\"t5-base\")\n",
    "        self.tokenizer = T5Tokenizer.from_pretrained(\"t5-base\", model_max_length=512)\n",
    "\n",
    "    def answer_question(self, question, context):\n",
    "        input_text = f\"question: {question} context: {context}\"\n",
    "        inputs = self.tokenizer(input_text, return_tensors=\"pt\", max_length=512, truncation=True)\n",
    "        answer_ids = self.model.generate(inputs.input_ids, max_length=150, num_beams=4, early_stopping=True)\n",
    "        return self.tokenizer.decode(answer_ids[0], skip_special_tokens=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db468f3d-b9c6-4d6a-8546-b135a252ef73",
   "metadata": {},
   "source": [
    "# Fungsi untuk Menanyakan Pertanyaan ke Abstrak yang Sudah Dirangkum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "484c7ff9-e2e3-4fd5-8a3c-9b1aecd13e9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ask_questions_on_abstracts(summaries, questions):\n",
    "    # Inisialisasi agen Question Answering\n",
    "    qa_agent = QuestionAnsweringAgent.remote()\n",
    "    \n",
    "    print(\"\\nJawaban Pertanyaan yang Tepat:\")\n",
    "    for i, summary in enumerate(summaries):\n",
    "        relevant_answers_found = False\n",
    "        # Mengganti \\n dalam ringkasan dengan line break asli\n",
    "        formatted_summary = summary.replace(\"\\\\n\", \"\\n\")\n",
    "        \n",
    "        # Tampilkan ringkasan yang sudah dirapikan\n",
    "        print(f\"\\nRingkasan Abstrak {i+1}:\\n{formatted_summary}\\n\")\n",
    "        \n",
    "        for question in questions:\n",
    "            answer = ray.get(qa_agent.answer_question.remote(question, formatted_summary))\n",
    "            # Hanya tampilkan jawaban yang relevan\n",
    "            if answer and answer.lower() not in [\"not\", \"jawaban tidak ditemukan\", \"not_entailment\"]:\n",
    "                if not relevant_answers_found:\n",
    "                    relevant_answers_found = True\n",
    "                print(f\"Pertanyaan: {question}\")\n",
    "                print(f\"Jawaban: {answer}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "015abfce-ca23-4439-ac20-12afe379678d",
   "metadata": {},
   "source": [
    "# Pipeline Utama dengan Pertanyaan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cada1387-912d-44c2-bde4-c2a33731abe9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def multi_agent_pipeline_with_questions(search_terms, user_query, folder_path=None, questions=[]):\n",
    "    # Menjalankan pipeline multi-agent sebelumnya untuk mencari, mengekstrak, dan merangkum abstrak\n",
    "    retrieval_agent = DataRetrievalAgent.remote()\n",
    "    extraction_agent = ExtractionAgent.remote()\n",
    "    summarization_agent = SummarizationAgent.remote()\n",
    "    \n",
    "    # Langkah 1: Cari PubMed berdasarkan kata kunci\n",
    "    pmids = ray.get(retrieval_agent.search_pubmed.remote(search_terms))\n",
    "    print(\"Daftar PMIDs yang ditemukan:\", pmids)\n",
    "    \n",
    "    # Langkah 2: Ekstraksi abstrak dari PubMed\n",
    "    abstracts = ray.get([extraction_agent.fetch_abstract.remote(pmid) for pmid in pmids])\n",
    "    print(\"Contoh abstrak:\", abstracts[0] if abstracts else \"Tidak ada abstrak ditemukan.\")\n",
    "    \n",
    "    # Langkah 3: Meringkas abstrak\n",
    "    summaries = ray.get([summarization_agent.summarize_text.remote(abstract) for abstract in abstracts])\n",
    "    \n",
    "    # Langkah 4: Menanyakan pertanyaan pada abstrak yang sudah diringkas\n",
    "    if questions:\n",
    "        ask_questions_on_abstracts(summaries, questions)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9a57552-0a64-4e92-ad21-1e4e0f5715b4",
   "metadata": {},
   "source": [
    "# Menjalankan Pipeline dengan Pertanyaan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd71770b-cba1-4251-ac28-c5df0b45c52a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[36m(SummarizationAgent pid=24901)\u001b[0m /opt/anaconda3/lib/python3.11/site-packages/huggingface_hub/file_download.py:1142: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.\n",
      "\u001b[36m(SummarizationAgent pid=24901)\u001b[0m   warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Daftar PMIDs yang ditemukan: ['39460909', '39460375', '39459574', '39459480', '39459455', '39460977', '39460909', '39460896', '39460887', '39460283']\n",
      "Contoh abstrak: 1. Diabetes Ther. 2024 Oct 26. doi: 10.1007/s13300-024-01658-8. Online ahead of \n",
      "print.\n",
      "\n",
      "Current Perspectives in Pre- and Diabetic Peripheral Neuropathy Diagnosis and \n",
      "Management: An Expert Statement for the Gulf Region.\n",
      "\n",
      "Beshyah SA(1)(2)(3), Jayyousi A(4), Al-Mamari AS(5), Shaaban A(6), Ozairi EA(7), \n",
      "Nafach J(8), Jallo MKI(9), Khader S(10), Evans M(11).\n",
      "\n",
      "Author information:\n",
      "(1)Department of Medicine, NMC Royal, MBZ, Abu Dhabi, United Arab Emirates. \n",
      "beshyah@yahoo.com.\n",
      "(2)Department of Medicine, College of Medicine and Health Sciences, Khalifa \n",
      "University, Abu Dhabi, UAE. beshyah@yahoo.com.\n",
      "(3)Department of Medicine, Dubai Medical College, Dubai, United Arab Emirates. \n",
      "beshyah@yahoo.com.\n",
      "(4)Hamad Medical Corporation, Doha, Qatar.\n",
      "(5)Department of Medicine, Sultan Qaboos University Hospital, Muscat, Oman.\n",
      "(6)Diabetes Control Centre, Ghassan Najib Pharaon Hospital, Jeddah, Kingdom of \n",
      "Saudi Arabia.\n",
      "(7)Clinical Care Research and Clinical Trials Unit, Dasman Diabetes Institute, \n",
      "Dasman, Kuwait.\n",
      "(8)Department of Diabetes, Dubai Diabetes Center, Dubai, United Arab Emirates.\n",
      "(9)Internal Medicine and Endocrinology, Centre of Endocrinology, Gulf Medical \n",
      "University and Thumbay University Hospital, Ajman, United Arab Emirates.\n",
      "(10)Department of Diabetes and Endocrinology, Dr. Sulaiman Alhabib Medical \n",
      "Group, Riyadh, Saudi Arabia.\n",
      "(11)Diabetes Resource Center, University Hospital Llandough, Penarth, UK.\n",
      "\n",
      "Peripheral neuropathy (PN) significantly impacts the quality of life, causing \n",
      "substantial morbidity and increased mortality, as well as escalating healthcare \n",
      "costs. While PN can have various causes, the most common form, diabetic \n",
      "peripheral neuropathy, poses considerable risks for potential complications. \n",
      "Diabetic peripheral neuropathy (DPN) affects over 50% of people with prediabetes \n",
      "and diabetes. Despite its prevalence, a global gap in diagnosis and management \n",
      "exists, exacerbated by the COVID-19 pandemic. This expert consensus was \n",
      "formulated through a comprehensive evaluation by a panel of experts, informed by \n",
      "a focused literature review, aiming to establish a clinically robust approach to \n",
      "diagnosing and managing pre- and diabetic PN with the early utilization of \n",
      "neurotropic B vitamins. This document offers a consensus perspective on the \n",
      "existing challenges in diagnosing and managing PN, focusing on DPN. The expert \n",
      "panel proposes measures to address this underdiagnosed burden, highlighting the \n",
      "importance of early intervention through innovative screening methods, \n",
      "integrated care approaches, and therapeutic strategies. The document advocates \n",
      "for increased awareness, targeted campaigns, and proactive care strategies to \n",
      "bridge gaps in the patient care of individuals with diabetes, emphasizing the \n",
      "importance of early detection and timely management to improve overall health \n",
      "outcomes. Specific recommendations include incorporating simplified \n",
      "questionnaires and innovative screening methods into routine care, prioritizing \n",
      "neurotropic B vitamin supplementation, optimizing glucagon-like peptide 1 \n",
      "(GLP-1) receptor agonist treatments, and adopting a holistic approach to \n",
      "neuropathy management. The consensus underscores the urgent need to address the \n",
      "underdiagnosis and undertreatment of PN, offering practical measures to enhance \n",
      "early detection and improve health outcomes for individuals with DPN.\n",
      "\n",
      "© 2024. The Author(s).\n",
      "\n",
      "DOI: 10.1007/s13300-024-01658-8\n",
      "PMID: 39460909\n"
     ]
    }
   ],
   "source": [
    "search_terms = [\"diabetes complications management\", \"innovations in diabetes treatment\"]\n",
    "user_query = \"What are the latest innovations in diabetes complications management?\"\n",
    "folder_path = \"/Users/dwiyulianto/Downloads/AI PROJECT/Journal_HHS Public Acces _Health\"\n",
    "questions = [\n",
    "    \"What is the impact of peripheral neuropathy on quality of life?\",\n",
    "    \"What are the common symptoms of diabetic peripheral neuropathy?\",\n",
    "]\n",
    "\n",
    "# Menjalankan pipeline dengan pertanyaan\n",
    "multi_agent_pipeline_with_questions(search_terms, user_query, folder_path, questions)\n",
    "\n",
    "# Shutdown Ray setelah selesai\n",
    "ray.shutdown()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eca21fdd-5bec-4190-8607-3fc320a1e77c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
